<!doctype html>
<html>
<head>
<meta charset='UTF-8'><meta name='viewport' content='width=device-width initial-scale=1'>
<title>gatk-pipeline-commands</title><style type='text/css'>html {overflow-x: initial !important;}:root { --bg-color:#ffffff; --text-color:#333333; --select-text-bg-color:#B5D6FC; --select-text-font-color:auto; --monospace:"Lucida Console",Consolas,"Courier",monospace; }
html { font-size: 14px; background-color: var(--bg-color); color: var(--text-color); font-family: "Helvetica Neue", Helvetica, Arial, sans-serif; -webkit-font-smoothing: antialiased; }
body { margin: 0px; padding: 0px; height: auto; bottom: 0px; top: 0px; left: 0px; right: 0px; font-size: 1rem; line-height: 1.42857; overflow-x: hidden; background: inherit; tab-size: 4; }
iframe { margin: auto; }
a.url { word-break: break-all; }
a:active, a:hover { outline: 0px; }
.in-text-selection, ::selection { text-shadow: none; background: var(--select-text-bg-color); color: var(--select-text-font-color); }
#write { margin: 0px auto; height: auto; width: inherit; word-break: normal; word-wrap: break-word; position: relative; white-space: normal; overflow-x: visible; padding-top: 40px; }
#write.first-line-indent p { text-indent: 2em; }
#write.first-line-indent li p, #write.first-line-indent p * { text-indent: 0px; }
#write.first-line-indent li { margin-left: 2em; }
.for-image #write { padding-left: 8px; padding-right: 8px; }
body.typora-export { padding-left: 30px; padding-right: 30px; }
.typora-export .footnote-line, .typora-export li, .typora-export p { white-space: pre-wrap; }
@media screen and (max-width: 500px) {
  body.typora-export { padding-left: 0px; padding-right: 0px; }
  #write { padding-left: 20px; padding-right: 20px; }
  .CodeMirror-sizer { margin-left: 0px !important; }
  .CodeMirror-gutters { display: none !important; }
}
#write li > figure:first-child { margin-top: -20px; }
#write ol, #write ul { position: relative; }
img { max-width: 100%; vertical-align: middle; }
button, input, select, textarea { color: inherit; font-style: inherit; font-variant: inherit; font-weight: inherit; font-stretch: inherit; font-size: inherit; line-height: inherit; font-family: inherit; }
input[type="checkbox"], input[type="radio"] { line-height: normal; padding: 0px; }
*, ::after, ::before { box-sizing: border-box; }
#write h1, #write h2, #write h3, #write h4, #write h5, #write h6, #write p, #write pre { width: inherit; }
#write h1, #write h2, #write h3, #write h4, #write h5, #write h6, #write p { position: relative; }
h1, h2, h3, h4, h5, h6 { break-after: avoid-page; break-inside: avoid; orphans: 2; }
p { orphans: 4; }
h1 { font-size: 2rem; }
h2 { font-size: 1.8rem; }
h3 { font-size: 1.6rem; }
h4 { font-size: 1.4rem; }
h5 { font-size: 1.2rem; }
h6 { font-size: 1rem; }
.md-math-block, .md-rawblock, h1, h2, h3, h4, h5, h6, p { margin-top: 1rem; margin-bottom: 1rem; }
.hidden { display: none; }
.md-blockmeta { color: rgb(204, 204, 204); font-weight: 700; font-style: italic; }
a { cursor: pointer; }
sup.md-footnote { padding: 2px 4px; background-color: rgba(238, 238, 238, 0.7); color: rgb(85, 85, 85); border-radius: 4px; cursor: pointer; }
sup.md-footnote a, sup.md-footnote a:hover { color: inherit; text-transform: inherit; text-decoration: inherit; }
#write input[type="checkbox"] { cursor: pointer; width: inherit; height: inherit; }
figure { overflow-x: auto; margin: 1.2em 0px; max-width: calc(100% + 16px); padding: 0px; }
figure > table { margin: 0px !important; }
tr { break-inside: avoid; break-after: auto; }
thead { display: table-header-group; }
table { border-collapse: collapse; border-spacing: 0px; width: 100%; overflow: auto; break-inside: auto; text-align: left; }
table.md-table td { min-width: 32px; }
.CodeMirror-gutters { border-right: 0px; background-color: inherit; }
.CodeMirror-linenumber { user-select: none; }
.CodeMirror { text-align: left; }
.CodeMirror-placeholder { opacity: 0.3; }
.CodeMirror pre { padding: 0px 4px; }
.CodeMirror-lines { padding: 0px; }
div.hr:focus { cursor: none; }
#write pre { white-space: pre-wrap; }
#write.fences-no-line-wrapping pre { white-space: pre; }
#write pre.ty-contain-cm { white-space: normal; }
.CodeMirror-gutters { margin-right: 4px; }
.md-fences { font-size: 0.9rem; display: block; break-inside: avoid; text-align: left; overflow: visible; white-space: pre; background: inherit; position: relative !important; }
.md-diagram-panel { width: 100%; margin-top: 10px; text-align: center; padding-top: 0px; padding-bottom: 8px; overflow-x: auto; }
#write .md-fences.mock-cm { white-space: pre-wrap; }
.md-fences.md-fences-with-lineno { padding-left: 0px; }
#write.fences-no-line-wrapping .md-fences.mock-cm { white-space: pre; overflow-x: auto; }
.md-fences.mock-cm.md-fences-with-lineno { padding-left: 8px; }
.CodeMirror-line, twitterwidget { break-inside: avoid; }
.footnotes { opacity: 0.8; font-size: 0.9rem; margin-top: 1em; margin-bottom: 1em; }
.footnotes + .footnotes { margin-top: 0px; }
.md-reset { margin: 0px; padding: 0px; border: 0px; outline: 0px; vertical-align: top; background: 0px 0px; text-decoration: none; text-shadow: none; float: none; position: static; width: auto; height: auto; white-space: nowrap; cursor: inherit; -webkit-tap-highlight-color: transparent; line-height: normal; font-weight: 400; text-align: left; box-sizing: content-box; direction: ltr; }
li div { padding-top: 0px; }
blockquote { margin: 1rem 0px; }
li .mathjax-block, li p { margin: 0.5rem 0px; }
li { margin: 0px; position: relative; }
blockquote > :last-child { margin-bottom: 0px; }
blockquote > :first-child, li > :first-child { margin-top: 0px; }
.footnotes-area { color: rgb(136, 136, 136); margin-top: 0.714rem; padding-bottom: 0.143rem; white-space: normal; }
#write .footnote-line { white-space: pre-wrap; }
@media print {
  body, html { border: 1px solid transparent; height: 99%; break-after: avoid; break-before: avoid; }
  #write { margin-top: 0px; padding-top: 0px; border-color: transparent !important; }
  .typora-export * { -webkit-print-color-adjust: exact; }
  html.blink-to-pdf { font-size: 13px; }
  .typora-export #write { padding-left: 32px; padding-right: 32px; padding-bottom: 0px; break-after: avoid; }
  .typora-export #write::after { height: 0px; }
  @page { margin: 20mm 0px; }
}
.footnote-line { margin-top: 0.714em; font-size: 0.7em; }
a img, img a { cursor: pointer; }
pre.md-meta-block { font-size: 0.8rem; min-height: 0.8rem; white-space: pre-wrap; background: rgb(204, 204, 204); display: block; overflow-x: hidden; }
p > .md-image:only-child:not(.md-img-error) img, p > img:only-child { display: block; margin: auto; }
p > .md-image:only-child { display: inline-block; width: 100%; }
#write .MathJax_Display { margin: 0.8em 0px 0px; }
.md-math-block { width: 100%; }
.md-math-block:not(:empty)::after { display: none; }
[contenteditable="true"]:active, [contenteditable="true"]:focus { outline: 0px; box-shadow: none; }
.md-task-list-item { position: relative; list-style-type: none; }
.task-list-item.md-task-list-item { padding-left: 0px; }
.md-task-list-item > input { position: absolute; top: 0px; left: 0px; margin-left: -1.2em; margin-top: calc(1em - 10px); border: none; }
.math { font-size: 1rem; }
.md-toc { min-height: 3.58rem; position: relative; font-size: 0.9rem; border-radius: 10px; }
.md-toc-content { position: relative; margin-left: 0px; }
.md-toc-content::after, .md-toc::after { display: none; }
.md-toc-item { display: block; color: rgb(65, 131, 196); }
.md-toc-item a { text-decoration: none; }
.md-toc-inner:hover { text-decoration: underline; }
.md-toc-inner { display: inline-block; cursor: pointer; }
.md-toc-h1 .md-toc-inner { margin-left: 0px; font-weight: 700; }
.md-toc-h2 .md-toc-inner { margin-left: 2em; }
.md-toc-h3 .md-toc-inner { margin-left: 4em; }
.md-toc-h4 .md-toc-inner { margin-left: 6em; }
.md-toc-h5 .md-toc-inner { margin-left: 8em; }
.md-toc-h6 .md-toc-inner { margin-left: 10em; }
@media screen and (max-width: 48em) {
  .md-toc-h3 .md-toc-inner { margin-left: 3.5em; }
  .md-toc-h4 .md-toc-inner { margin-left: 5em; }
  .md-toc-h5 .md-toc-inner { margin-left: 6.5em; }
  .md-toc-h6 .md-toc-inner { margin-left: 8em; }
}
a.md-toc-inner { font-size: inherit; font-style: inherit; font-weight: inherit; line-height: inherit; }
.footnote-line a:not(.reversefootnote) { color: inherit; }
.md-attr { display: none; }
.md-fn-count::after { content: "."; }
code, pre, samp, tt { font-family: var(--monospace); }
kbd { margin: 0px 0.1em; padding: 0.1em 0.6em; font-size: 0.8em; color: rgb(36, 39, 41); background: rgb(255, 255, 255); border: 1px solid rgb(173, 179, 185); border-radius: 3px; box-shadow: rgba(12, 13, 14, 0.2) 0px 1px 0px, rgb(255, 255, 255) 0px 0px 0px 2px inset; white-space: nowrap; vertical-align: middle; }
.md-comment { color: rgb(162, 127, 3); opacity: 0.8; font-family: var(--monospace); }
code { text-align: left; vertical-align: initial; }
a.md-print-anchor { white-space: pre !important; border-width: initial !important; border-style: none !important; border-color: initial !important; display: inline-block !important; position: absolute !important; width: 1px !important; right: 0px !important; outline: 0px !important; background: 0px 0px !important; text-decoration: initial !important; text-shadow: initial !important; }
.md-inline-math .MathJax_SVG .noError { display: none !important; }
.html-for-mac .inline-math-svg .MathJax_SVG { vertical-align: 0.2px; }
.md-math-block .MathJax_SVG_Display { text-align: center; margin: 0px; position: relative; text-indent: 0px; max-width: none; max-height: none; min-height: 0px; min-width: 100%; width: auto; overflow-y: hidden; display: block !important; }
.MathJax_SVG_Display, .md-inline-math .MathJax_SVG_Display { width: auto; margin: inherit; display: inline-block !important; }
.MathJax_SVG .MJX-monospace { font-family: var(--monospace); }
.MathJax_SVG .MJX-sans-serif { font-family: sans-serif; }
.MathJax_SVG { display: inline; font-style: normal; font-weight: 400; line-height: normal; zoom: 90%; text-indent: 0px; text-align: left; text-transform: none; letter-spacing: normal; word-spacing: normal; word-wrap: normal; white-space: nowrap; float: none; direction: ltr; max-width: none; max-height: none; min-width: 0px; min-height: 0px; border: 0px; padding: 0px; margin: 0px; }
.MathJax_SVG * { transition: none; }
.MathJax_SVG_Display svg { vertical-align: middle !important; margin-bottom: 0px !important; }
.os-windows.monocolor-emoji .md-emoji { font-family: "Segoe UI Symbol", sans-serif; }
.md-diagram-panel > svg { max-width: 100%; }
[lang="mermaid"] svg, [lang="flow"] svg { max-width: 100%; }
[lang="mermaid"] .node text { font-size: 1rem; }
table tr th { border-bottom: 0px; }
video { max-width: 100%; display: block; margin: 0px auto; }
iframe { max-width: 100%; width: 100%; border: none; }
.highlight td, .highlight tr { border: 0px; }


.CodeMirror { height: auto; }
.CodeMirror.cm-s-inner { background: inherit; }
.CodeMirror-scroll { overflow-y: hidden; overflow-x: auto; z-index: 3; }
.CodeMirror-gutter-filler, .CodeMirror-scrollbar-filler { background-color: rgb(255, 255, 255); }
.CodeMirror-gutters { border-right: 1px solid rgb(221, 221, 221); background: inherit; white-space: nowrap; }
.CodeMirror-linenumber { padding: 0px 3px 0px 5px; text-align: right; color: rgb(153, 153, 153); }
.cm-s-inner .cm-keyword { color: rgb(119, 0, 136); }
.cm-s-inner .cm-atom, .cm-s-inner.cm-atom { color: rgb(34, 17, 153); }
.cm-s-inner .cm-number { color: rgb(17, 102, 68); }
.cm-s-inner .cm-def { color: rgb(0, 0, 255); }
.cm-s-inner .cm-variable { color: rgb(0, 0, 0); }
.cm-s-inner .cm-variable-2 { color: rgb(0, 85, 170); }
.cm-s-inner .cm-variable-3 { color: rgb(0, 136, 85); }
.cm-s-inner .cm-string { color: rgb(170, 17, 17); }
.cm-s-inner .cm-property { color: rgb(0, 0, 0); }
.cm-s-inner .cm-operator { color: rgb(152, 26, 26); }
.cm-s-inner .cm-comment, .cm-s-inner.cm-comment { color: rgb(170, 85, 0); }
.cm-s-inner .cm-string-2 { color: rgb(255, 85, 0); }
.cm-s-inner .cm-meta { color: rgb(85, 85, 85); }
.cm-s-inner .cm-qualifier { color: rgb(85, 85, 85); }
.cm-s-inner .cm-builtin { color: rgb(51, 0, 170); }
.cm-s-inner .cm-bracket { color: rgb(153, 153, 119); }
.cm-s-inner .cm-tag { color: rgb(17, 119, 0); }
.cm-s-inner .cm-attribute { color: rgb(0, 0, 204); }
.cm-s-inner .cm-header, .cm-s-inner.cm-header { color: rgb(0, 0, 255); }
.cm-s-inner .cm-quote, .cm-s-inner.cm-quote { color: rgb(0, 153, 0); }
.cm-s-inner .cm-hr, .cm-s-inner.cm-hr { color: rgb(153, 153, 153); }
.cm-s-inner .cm-link, .cm-s-inner.cm-link { color: rgb(0, 0, 204); }
.cm-negative { color: rgb(221, 68, 68); }
.cm-positive { color: rgb(34, 153, 34); }
.cm-header, .cm-strong { font-weight: 700; }
.cm-del { text-decoration: line-through; }
.cm-em { font-style: italic; }
.cm-link { text-decoration: underline; }
.cm-error { color: red; }
.cm-invalidchar { color: red; }
.cm-constant { color: rgb(38, 139, 210); }
.cm-defined { color: rgb(181, 137, 0); }
div.CodeMirror span.CodeMirror-matchingbracket { color: rgb(0, 255, 0); }
div.CodeMirror span.CodeMirror-nonmatchingbracket { color: rgb(255, 34, 34); }
.cm-s-inner .CodeMirror-activeline-background { background: inherit; }
.CodeMirror { position: relative; overflow: hidden; }
.CodeMirror-scroll { height: 100%; outline: 0px; position: relative; box-sizing: content-box; background: inherit; }
.CodeMirror-sizer { position: relative; }
.CodeMirror-gutter-filler, .CodeMirror-hscrollbar, .CodeMirror-scrollbar-filler, .CodeMirror-vscrollbar { position: absolute; z-index: 6; display: none; }
.CodeMirror-vscrollbar { right: 0px; top: 0px; overflow: hidden; }
.CodeMirror-hscrollbar { bottom: 0px; left: 0px; overflow: hidden; }
.CodeMirror-scrollbar-filler { right: 0px; bottom: 0px; }
.CodeMirror-gutter-filler { left: 0px; bottom: 0px; }
.CodeMirror-gutters { position: absolute; left: 0px; top: 0px; padding-bottom: 30px; z-index: 3; }
.CodeMirror-gutter { white-space: normal; height: 100%; box-sizing: content-box; padding-bottom: 30px; margin-bottom: -32px; display: inline-block; }
.CodeMirror-gutter-wrapper { position: absolute; z-index: 4; background: 0px 0px !important; border: none !important; }
.CodeMirror-gutter-background { position: absolute; top: 0px; bottom: 0px; z-index: 4; }
.CodeMirror-gutter-elt { position: absolute; cursor: default; z-index: 4; }
.CodeMirror-lines { cursor: text; }
.CodeMirror pre { border-radius: 0px; border-width: 0px; background: 0px 0px; font-family: inherit; font-size: inherit; margin: 0px; white-space: pre; word-wrap: normal; color: inherit; z-index: 2; position: relative; overflow: visible; }
.CodeMirror-wrap pre { word-wrap: break-word; white-space: pre-wrap; word-break: normal; }
.CodeMirror-code pre { border-right: 30px solid transparent; width: fit-content; }
.CodeMirror-wrap .CodeMirror-code pre { border-right: none; width: auto; }
.CodeMirror-linebackground { position: absolute; left: 0px; right: 0px; top: 0px; bottom: 0px; z-index: 0; }
.CodeMirror-linewidget { position: relative; z-index: 2; overflow: auto; }
.CodeMirror-wrap .CodeMirror-scroll { overflow-x: hidden; }
.CodeMirror-measure { position: absolute; width: 100%; height: 0px; overflow: hidden; visibility: hidden; }
.CodeMirror-measure pre { position: static; }
.CodeMirror div.CodeMirror-cursor { position: absolute; visibility: hidden; border-right: none; width: 0px; }
.CodeMirror div.CodeMirror-cursor { visibility: hidden; }
.CodeMirror-focused div.CodeMirror-cursor { visibility: inherit; }
.cm-searching { background: rgba(255, 255, 0, 0.4); }
@media print {
  .CodeMirror div.CodeMirror-cursor { visibility: hidden; }
}


html { font-size: 16px; }
body { background: rgb(255, 255, 255); font-family: "Source Han Sans SC", sans-serif; color: rgb(85, 85, 85); }
#write { max-width: 60em; }
h1, h2, h3, h4, h5, h6 { position: relative; margin-top: 1em; margin-bottom: 1em; font-weight: bold; line-height: 1.4em; }
h1 { font-size: 2em; line-height: 1.2em; }
h2 { font-size: 1.5em; line-height: 1.225em; }
h3 { font-size: 1.3em; line-height: 1.43em; }
h4 { font-size: 1.2em; }
h5 { font-size: 1em; }
h6 { font-size: 1em; color: rgb(153, 153, 153); }
hr { border: 1px solid rgb(221, 221, 221); }
a { text-decoration: none; color: rgb(37, 143, 184); }
a:hover, a:active { text-decoration: underline; }
ul, ol { padding-left: 2em; }
.task-list { padding-left: 2em; }
.task-list-item { padding-left: 1.6em; }
.task-list-item input { top: 3px; left: 8px; }
table { width: 100%; border-collapse: collapse; border-spacing: 0px; }
table th { font-weight: bold; border-bottom: 3px solid rgb(221, 221, 221); padding-bottom: 0.5em; }
table td { border-bottom: 1px solid rgb(221, 221, 221); padding: 10px 0px; }
blockquote { border-left: 5px solid rgb(221, 221, 221); padding-left: 0.5em; font-family: "Source Han Serif SC", serif; color: rgb(85, 85, 85); }
blockquote blockquote { padding-right: 0px; }
.md-fences, code, tt { margin: 0px 0.3em; padding: 0px 0.3em; background: rgb(238, 238, 238); font-family: mononoki, monospace; text-shadow: rgb(255, 255, 255) 0px 1px; }
.md-fences { margin: 15px auto; padding: 0.7em 1em; text-shadow: none; }
.cm-s-inner { background-color: rgb(38, 50, 56); color: rgb(233, 237, 237); }
.cm-s-inner .CodeMirror-gutters { background: rgb(38, 50, 56); color: rgb(83, 127, 126); border: none; }
.cm-s-inner .CodeMirror-guttermarker, .cm-s-inner .CodeMirror-guttermarker-subtle, .cm-s-inner .CodeMirror-linenumber { color: rgb(83, 127, 126); }
.cm-s-inner .CodeMirror-cursor { border-left: 1px solid rgb(248, 248, 240); }
.cm-s-inner div.CodeMirror-selected { background: rgba(255, 255, 255, 0.15); }
.cm-s-inner.CodeMirror-focused div.CodeMirror-selected { background: rgba(255, 255, 255, 0.1); }
.cm-s-inner .CodeMirror-line::selection, .cm-s-inner .CodeMirror-line > span::selection, .cm-s-inner .CodeMirror-line > span > span::selection { background: rgba(255, 255, 255, 0.1); }
.cm-s-inner .CodeMirror-activeline-background { background: rgba(0, 0, 0, 0); }
.cm-s-inner .cm-keyword { color: rgb(199, 146, 234); }
.cm-s-inner .cm-operator { color: rgb(233, 237, 237); }
.cm-s-inner .cm-variable-2 { color: rgb(128, 203, 196); }
.cm-s-inner .cm-variable-3 { color: rgb(130, 177, 255); }
.cm-s-inner .cm-builtin { color: rgb(222, 203, 107); }
.cm-s-inner .cm-atom { color: rgb(247, 118, 105); }
.cm-s-inner .cm-number { color: rgb(247, 118, 105); }
.cm-s-inner .cm-def { color: rgb(233, 237, 237); }
.cm-s-inner .cm-string { color: rgb(195, 232, 141); }
.cm-s-inner .cm-string-2 { color: rgb(128, 203, 196); }
.cm-s-inner .cm-comment { color: rgb(84, 110, 122); }
.cm-s-inner .cm-variable { color: rgb(130, 177, 255); }
.cm-s-inner .cm-tag { color: rgb(128, 203, 196); }
.cm-s-inner .cm-meta { color: rgb(128, 203, 196); }
.cm-s-inner .cm-attribute { color: rgb(255, 203, 107); }
.cm-s-inner .cm-property { color: rgb(128, 203, 174); }
.cm-s-inner .cm-qualifier { color: rgb(222, 203, 107); }
.cm-s-inner .cm-variable-3 { color: rgb(222, 203, 107); }
.cm-s-inner .cm-tag { color: rgb(255, 83, 112); }
.cm-s-inner .cm-error { color: rgb(255, 255, 255); background-color: rgb(236, 95, 103); }
.cm-s-inner .CodeMirror-matchingbracket { text-decoration: underline; color: white !important; }
.md-fences { background-color: rgb(38, 50, 56); color: rgb(233, 237, 237); border: none; }
.md-fences .code-tooltip { background-color: rgb(38, 50, 56); }
.cm-s-typora-default { background-color: rgb(38, 50, 56); color: rgb(233, 237, 237); }
.cm-s-typora-default .CodeMirror-gutters { background: rgb(38, 50, 56); color: rgb(83, 127, 126); border: none; }
.cm-s-typora-default .CodeMirror-guttermarker, .cm-s-typora-default .CodeMirror-guttermarker-subtle, .cm-s-typora-default .CodeMirror-linenumber { color: rgb(83, 127, 126); }
.cm-s-typora-default .CodeMirror-cursor { border-left: 1px solid rgb(248, 248, 240); }
.cm-s-typora-default div.CodeMirror-selected { background: rgba(255, 255, 255, 0.15); }
.cm-s-typora-default.CodeMirror-focused div.CodeMirror-selected { background: rgba(255, 255, 255, 0.1); }
.cm-s-typora-default .CodeMirror-line::selection, .cm-s-typora-default .CodeMirror-line > span::selection, .cm-s-typora-default .CodeMirror-line > span > span::selection { background: rgba(255, 255, 255, 0.1); }
.cm-s-typora-default .CodeMirror-activeline-background { background: rgba(0, 0, 0, 0); }
.cm-s-typora-default .cm-keyword { color: rgb(199, 146, 234); }
.cm-s-typora-default .cm-operator { color: rgb(233, 237, 237); }
.cm-s-typora-default .cm-variable-2 { color: rgb(128, 203, 196); }
.cm-s-typora-default .cm-variable-3 { color: rgb(130, 177, 255); }
.cm-s-typora-default .cm-builtin { color: rgb(222, 203, 107); }
.cm-s-typora-default .cm-atom { color: rgb(247, 118, 105); }
.cm-s-typora-default .cm-number { color: rgb(247, 118, 105); }
.cm-s-typora-default .cm-def { color: rgb(233, 237, 237); }
.cm-s-typora-default .cm-string { color: rgb(195, 232, 141); }
.cm-s-typora-default .cm-string-2 { color: rgb(128, 203, 196); }
.cm-s-typora-default .cm-comment { color: rgb(84, 110, 122); }
.cm-s-typora-default .cm-variable { color: rgb(130, 177, 255); }
.cm-s-typora-default .cm-tag { color: rgb(128, 203, 196); }
.cm-s-typora-default .cm-meta { color: rgb(128, 203, 196); }
.cm-s-typora-default .cm-attribute { color: rgb(255, 203, 107); }
.cm-s-typora-default .cm-property { color: rgb(128, 203, 174); }
.cm-s-typora-default .cm-qualifier { color: rgb(222, 203, 107); }
.cm-s-typora-default .cm-variable-3 { color: rgb(222, 203, 107); }
.cm-s-typora-default .cm-tag { color: rgb(255, 83, 112); }
.cm-s-typora-default .cm-error { color: rgb(255, 255, 255); background-color: rgb(236, 95, 103); }
.cm-s-typora-default .CodeMirror-matchingbracket { text-decoration: underline; color: white !important; }
.md-fences { background-color: rgb(38, 50, 56); color: rgb(233, 237, 237); border: none; }
.md-fences .code-tooltip { background-color: rgb(38, 50, 56); }
.CodeMirror div.CodeMirror-cursor { border-left: 1px solid rgb(228, 98, 154); }
#write pre.md-meta-block { padding: 1em; background-color: rgb(255, 255, 255); border-bottom: 1px dashed rgb(221, 221, 221); color: rgb(128, 203, 196); margin-top: 0px !important; }
.md-image > .md-meta { color: inherit; }
header, .context-menu, .megamenu-content, footer { font-family: "Source Han Sans SC", sans-serif; }
@media print {
  html { font-size: 13px; }
  table, pre { break-inside: avoid; }
  pre { word-wrap: break-word; }
}
header, .context-menu, .megamenu-content, footer { font-family: "Helvetica Neue", Helvetica, "Segoe UI", Arial, sans-serif; }
.md-diagram-panel-preview { color: rgb(38, 50, 56); }





 .typora-export li, .typora-export p, .typora-export,  .footnote-line {white-space: normal;} 
</style>
</head>
<body class='typora-export' >
<div  id='write'  class = 'is-node'><h1><a name='header-n250' class='md-header-anchor '></a>gatk pipeline with snakemake</h1><p>Output of the following command have been annotated</p><pre spellcheck="false" class="md-fences md-end-block ty-contain-cm modeLoaded" lang="bash"><div class="CodeMirror cm-s-inner CodeMirror-wrap" lang="bash"><div style="overflow: hidden; position: relative; width: 3px; height: 0px; top: 0px; left: 4.16669px;"><textarea autocorrect="off" autocapitalize="off" spellcheck="false" tabindex="0" style="position: absolute; bottom: -1em; padding: 0px; width: 1000px; height: 1em; outline: none;"></textarea></div><div class="CodeMirror-scrollbar-filler" cm-not-content="true"></div><div class="CodeMirror-gutter-filler" cm-not-content="true"></div><div class="CodeMirror-scroll" tabindex="-1"><div class="CodeMirror-sizer" style="margin-left: 0px; margin-bottom: 0px; border-right-width: 0px; padding-right: 0px; padding-bottom: 0px;"><div style="position: relative; top: 0px;"><div class="CodeMirror-lines" role="presentation"><div role="presentation" style="position: relative; outline: none;"><div class="CodeMirror-measure"><pre><span>xxxxxxxxxx</span></pre></div><div class="CodeMirror-measure"></div><div style="position: relative; z-index: 1;"></div><div class="CodeMirror-code" role="presentation"><div class="CodeMirror-activeline" style="position: relative;"><div class="CodeMirror-activeline-background CodeMirror-linebackground"></div><div class="CodeMirror-gutter-background CodeMirror-activeline-gutter" style="left: 0px; width: 0px;"></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">snakemake <span class="cm-attribute">-p</span> <span class="cm-attribute">--use-conda</span> <span class="cm-attribute">--cores</span> <span class="cm-number">8</span> &gt; gatk-pipeline-commands <span class="cm-number">2</span>&gt;&amp;1</span></pre></div></div></div></div></div></div><div style="position: absolute; height: 0px; width: 1px; border-bottom: 0px solid transparent; top: 20px;"></div><div class="CodeMirror-gutters" style="display: none; height: 20px;"></div></div></div></pre><h2><a name='header-n254' class='md-header-anchor '></a>Info</h2><p>/home/alper/miniconda3/lib/python3.7/site-packages/snakemake/workflow.py:12: FutureWarning: read_table is deprecated, use read_csv instead, passing sep=&#39;\t&#39;.
  from collections import OrderedDict
/home/alper/miniconda3/lib/python3.7/site-packages/snakemake/workflow.py:15: FutureWarning: read_table is deprecated, use read_csv instead, passing sep=&#39;\t&#39;.
  from operator import attrgetter
/home/alper/miniconda3/lib/python3.7/site-packages/snakemake/workflow.py:21: FutureWarning: read_table is deprecated, use read_csv instead, passing sep=&#39;\t&#39;.
  from snakemake.exceptions import RuleException, CreateRuleException, \
Building DAG of jobs...
WARNING: The conda.compat module is deprecated and will be removed in a future release.
WARNING: The conda.compat module is deprecated and will be removed in a future release.
Using shell: /bin/bash
Provided cores: 8
Rules claiming more threads will be scaled down.
Job counts:
<span>	</span>count<span>	</span>jobs
<span>	</span>1<span>	</span>all
<span>	</span>2<span>	</span>call_variants
<span>	</span>1<span>	</span>combine_calls
<span>	</span>3<span>	</span>fastqc
<span>	</span>1<span>	</span>genotype_variants
<span>	</span>2<span>	</span>hard_filter_calls
<span>	</span>3<span>	</span>map_reads
<span>	</span>3<span>	</span>mark_duplicates
<span>	</span>1<span>	</span>merge_calls
<span>	</span>1<span>	</span>merge_variants
<span>	</span>1<span>	</span>multiqc
<span>	</span>1<span>	</span>plot_stats
<span>	</span>3<span>	</span>recalibrate_base_qualities
<span>	</span>3<span>	</span>samtools_stats
<span>	</span>2<span>	</span>select_calls
<span>	</span>1<span>	</span>snpeff
<span>	</span>2<span>	</span>trim_reads_pe
<span>	</span>1<span>	</span>trim_reads_se
<span>	</span>1<span>	</span>vcf_to_tsv
<span>	</span>33</p><h2><a name='header-n256' class='md-header-anchor '></a>job1 : trim_reads_se</h2><p>&nbsp;</p><p>[Mon Apr 29 10:37:26 2019]
rule trim_reads_se:
    input: data/reads/b.chr21.1.fq
    output: trimmed/B-2.fastq.gz
    log: logs/trimmomatic/B-2.log
    jobid: 27
    wildcards: sample=B, unit=2</p><p>[Mon Apr 29 10:37:26 2019]
rule trim_reads_pe:
    input: data/reads/b.chr21.1.fq, data/reads/b.chr21.2.fq
    output: trimmed/B-1.1.fastq.gz, trimmed/B-1.2.fastq.gz, trimmed/B-1.1.unpaired.fastq.gz, trimmed/B-1.2.unpaired.fastq.gz, trimmed/B-1.trimlog.txt
    log: logs/trimmomatic/B-1.log
    jobid: 26
    wildcards: sample=B, unit=1</p><p>[Mon Apr 29 10:37:26 2019]
rule trim_reads_pe:
    input: data/reads/a.chr21.1.fq, data/reads/a.chr21.2.fq
    output: trimmed/A-1.1.fastq.gz, trimmed/A-1.2.fastq.gz, trimmed/A-1.1.unpaired.fastq.gz, trimmed/A-1.2.unpaired.fastq.gz, trimmed/A-1.trimlog.txt
    log: logs/trimmomatic/A-1.log
    jobid: 25
    wildcards: sample=A, unit=1</p><p>[Mon Apr 29 10:37:26 2019]
rule fastqc:
    input: data/reads/b.chr21.1.fq
    output: qc/fastqc/B-2.html, qc/fastqc/B-2.zip
    jobid: 12
    wildcards: sample=B, unit=2</p><p>[Mon Apr 29 10:37:26 2019]
rule fastqc:
    input: data/reads/b.chr21.1.fq, data/reads/b.chr21.2.fq
    output: qc/fastqc/B-1.html, qc/fastqc/B-1.zip
    jobid: 9
    wildcards: sample=B, unit=1</p><p>[Mon Apr 29 10:37:26 2019]
rule fastqc:
    input: data/reads/a.chr21.1.fq, data/reads/a.chr21.2.fq
    output: qc/fastqc/A-1.html, qc/fastqc/A-1.zip
    jobid: 6
    wildcards: sample=A, unit=1</p><p>/home/alper/miniconda3/bin/python3.7 /home/alper/tmp/snakemake-dna-seq-gatk-variant-calling-1.0.1/.snakemake/scripts/tmpdu7k4w0<em>.wrapper.py
/home/alper/miniconda3/bin/python3.7 /home/alper/tmp/snakemake-dna-seq-gatk-variant-calling-1.0.1/.snakemake/scripts/tmp3m4byz_e.wrapper.py
/home/alper/miniconda3/bin/python3.7 /home/alper/tmp/snakemake-dna-seq-gatk-variant-calling-1.0.1/.snakemake/scripts/tmpu9u4sqgl.wrapper.py
/home/alper/miniconda3/bin/python3.7 /home/alper/tmp/snakemake-dna-seq-gatk-variant-calling-1.0.1/.snakemake/scripts/tmpqczx5dpr.wrapper.py
/home/alper/miniconda3/bin/python3.7 /home/alper/tmp/snakemake-dna-seq-gatk-variant-calling-1.0.1/.snakemake/scripts/tmp8nr4i52y.wrapper.py
/home/alper/miniconda3/bin/python3.7 /home/alper/tmp/snakemake-dna-seq-gatk-variant-calling-1.0.1/.snakemake/scripts/tmpt1scdsu</em>.wrapper.py
WARNING: The conda.compat module is deprecated and will be removed in a future release.
WARNING: The conda.compat module is deprecated and will be removed in a future release.
Activating conda environment: /home/alper/tmp/snakemake-dna-seq-gatk-variant-calling-1.0.1/.snakemake/conda/c0b8860d
Activating conda environment: /home/alper/tmp/snakemake-dna-seq-gatk-variant-calling-1.0.1/.snakemake/conda/c0b8860d
WARNING: The conda.compat module is deprecated and will be removed in a future release.
WARNING: The conda.compat module is deprecated and will be removed in a future release.
WARNING: The conda.compat module is deprecated and will be removed in a future release.
WARNING: The conda.compat module is deprecated and will be removed in a future release.
Activating conda environment: /home/alper/tmp/snakemake-dna-seq-gatk-variant-calling-1.0.1/.snakemake/conda/c0b8860d
Activating conda environment: /home/alper/tmp/snakemake-dna-seq-gatk-variant-calling-1.0.1/.snakemake/conda/e0139b03
Activating conda environment: /home/alper/tmp/snakemake-dna-seq-gatk-variant-calling-1.0.1/.snakemake/conda/e0139b03
Activating conda environment: /home/alper/tmp/snakemake-dna-seq-gatk-variant-calling-1.0.1/.snakemake/conda/e0139b03</p><pre spellcheck="false" class="md-fences md-end-block ty-contain-cm modeLoaded" lang=""><div class="CodeMirror cm-s-inner CodeMirror-wrap" lang=""><div style="overflow: hidden; position: relative; width: 3px; height: 0px; top: 0px; left: 4.16669px;"><textarea autocorrect="off" autocapitalize="off" spellcheck="false" tabindex="0" style="position: absolute; bottom: -1em; padding: 0px; width: 1000px; height: 1em; outline: none;"></textarea></div><div class="CodeMirror-scrollbar-filler" cm-not-content="true"></div><div class="CodeMirror-gutter-filler" cm-not-content="true"></div><div class="CodeMirror-scroll" tabindex="-1"><div class="CodeMirror-sizer" style="margin-left: 0px; margin-bottom: 0px; border-right-width: 0px; padding-right: 0px; padding-bottom: 0px;"><div style="position: relative; top: 0px;"><div class="CodeMirror-lines" role="presentation"><div role="presentation" style="position: relative; outline: none;"><div class="CodeMirror-measure"><pre><span>xxxxxxxxxx</span></pre></div><div class="CodeMirror-measure"></div><div style="position: relative; z-index: 1;"></div><div class="CodeMirror-code" role="presentation" style=""><div class="CodeMirror-activeline" style="position: relative;"><div class="CodeMirror-activeline-background CodeMirror-linebackground"></div><div class="CodeMirror-gutter-background CodeMirror-activeline-gutter" style="left: 0px; width: 0px;"></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">trimmomatic PE -trimlog trimmed/A-1.trimlog.txt data/reads/a.chr21.1.fq data/reads/a.chr21.2.fq &gt;(pigz -5 &gt; trimmed/A-1.1.fastq.gz) &gt;(pigz -5 &gt; trimmed/A-1.1.unpaired.fastq.gz) &gt;(pigz -5 &gt; trimmed/A-1.2.fastq.gz) &gt;(pigz -5 &gt; trimmed/A-1.2.unpaired.fastq.gz) LEADING:3 TRAILING:3 SLIDINGWINDOW:4:15 MINLEN:36  &gt; logs/trimmomatic/A-1.log 2&gt;&amp;1</span></pre></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="">​</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">trimmomatic SE  data/reads/b.chr21.1.fq &gt;(pigz -5 &gt; trimmed/B-2.fastq.gz) LEADING:3 TRAILING:3 SLIDINGWINDOW:4:15 MINLEN:36  &gt; logs/trimmomatic/B-2.log 2&gt;&amp;1</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">fastqc  --quiet --outdir /tmp/tmp8a3c359i data/reads/b.chr21.1.fq </span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="">​</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">trimmomatic PE -trimlog trimmed/B-1.trimlog.txt data/reads/b.chr21.1.fq data/reads/b.chr21.2.fq &gt;(pigz -5 &gt; trimmed/B-1.1.fastq.gz) &gt;(pigz -5 &gt; trimmed/B-1.1.unpaired.fastq.gz) &gt;(pigz -5 &gt; trimmed/B-1.2.fastq.gz) &gt;(pigz -5 &gt; trimmed/B-1.2.unpaired.fastq.gz) LEADING:3 TRAILING:3 SLIDINGWINDOW:4:15 MINLEN:36  &gt; logs/trimmomatic/B-1.log 2&gt;&amp;1</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="">​</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">fastqc  --quiet --outdir /tmp/tmp902azgfu data/reads/b.chr21.1.fq </span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">fastqc  --quiet --outdir /tmp/tmpaqjqld16 data/reads/a.chr21.1.fq </span></pre></div></div></div></div></div><div style="position: absolute; height: 0px; width: 1px; border-bottom: 0px solid transparent; top: 320px;"></div><div class="CodeMirror-gutters" style="display: none; height: 320px;"></div></div></div></pre><p>[Mon Apr 29 10:37:29 2019]
Finished job 27.
1 of 33 steps (3%) done
Removing temporary output file trimmed/A-1.1.unpaired.fastq.gz.
Removing temporary output file trimmed/A-1.2.unpaired.fastq.gz.
[Mon Apr 29 10:37:31 2019]
Finished job 25.
2 of 33 steps (6%) done
Removing temporary output file trimmed/B-1.1.unpaired.fastq.gz.
Removing temporary output file trimmed/B-1.2.unpaired.fastq.gz.
[Mon Apr 29 10:37:31 2019]
Finished job 26.
3 of 33 steps (9%) done
mv /tmp/tmp8a3c359i/b.chr21.1_fastqc.html qc/fastqc/B-1.html
mv /tmp/tmp8a3c359i/b.chr21.1_fastqc.zip qc/fastqc/B-1.zip
[Mon Apr 29 10:37:33 2019]
Finished job 9.
4 of 33 steps (12%) done
mv /tmp/tmp902azgfu/b.chr21.1_fastqc.html qc/fastqc/B-2.html
mv /tmp/tmp902azgfu/b.chr21.1_fastqc.zip qc/fastqc/B-2.zip
[Mon Apr 29 10:37:33 2019]
Finished job 12.
5 of 33 steps (15%) done
mv /tmp/tmpaqjqld16/a.chr21.1_fastqc.html qc/fastqc/A-1.html
mv /tmp/tmpaqjqld16/a.chr21.1_fastqc.zip qc/fastqc/A-1.zip
[Mon Apr 29 10:37:33 2019]
Finished job 6.
6 of 33 steps (18%) done</p><p>[Mon Apr 29 10:37:29 2019]
Finished job 27.
1 of 33 steps (3%) done
Removing temporary output file trimmed/A-1.1.unpaired.fastq.gz.
Removing temporary output file trimmed/A-1.2.unpaired.fastq.gz.
[Mon Apr 29 10:37:31 2019]
Finished job 25.
2 of 33 steps (6%) done
Removing temporary output file trimmed/B-1.1.unpaired.fastq.gz.
Removing temporary output file trimmed/B-1.2.unpaired.fastq.gz.
[Mon Apr 29 10:37:31 2019]
Finished job 26.
3 of 33 steps (9%) done
mv /tmp/tmp8a3c359i/b.chr21.1_fastqc.html qc/fastqc/B-1.html
mv /tmp/tmp8a3c359i/b.chr21.1_fastqc.zip qc/fastqc/B-1.zip
[Mon Apr 29 10:37:33 2019]
Finished job 9.
4 of 33 steps (12%) done
mv /tmp/tmp902azgfu/b.chr21.1_fastqc.html qc/fastqc/B-2.html
mv /tmp/tmp902azgfu/b.chr21.1_fastqc.zip qc/fastqc/B-2.zip
[Mon Apr 29 10:37:33 2019]
Finished job 12.
5 of 33 steps (15%) done
mv /tmp/tmpaqjqld16/a.chr21.1_fastqc.html qc/fastqc/A-1.html
mv /tmp/tmpaqjqld16/a.chr21.1_fastqc.zip qc/fastqc/A-1.zip
[Mon Apr 29 10:37:33 2019]
Finished job 6.
6 of 33 steps (18%) done</p><p>[Mon Apr 29 10:37:33 2019]
rule map_reads:
    input: trimmed/B-1.1.fastq.gz, trimmed/B-1.2.fastq.gz
    output: mapped/B-1.sorted.bam
    log: logs/bwa_mem/B-1.log
    jobid: 20
    wildcards: sample=B, unit=1
    threads: 8</p><p>python /home/alper/tmp/snakemake-dna-seq-gatk-variant-calling-1.0.1/.snakemake/scripts/tmpmqx3dygi.wrapper.py
WARNING: The conda.compat module is deprecated and will be removed in a future release.
Activating conda environment: /home/alper/tmp/snakemake-dna-seq-gatk-variant-calling-1.0.1/.snakemake/conda/332db904</p><p>Removing temporary output file trimmed/B-1.2.fastq.gz.
Removing temporary output file trimmed/B-1.1.fastq.gz.
[Mon Apr 29 10:37:36 2019]
Finished job 20.
7 of 33 steps (21%) done</p><p>[Mon Apr 29 10:37:36 2019]
rule mark_duplicates:
    input: mapped/B-1.sorted.bam
    output: dedup/B-1.bam, qc/dedup/B-1.metrics.txt
    log: logs/picard/dedup/B-1.log
    jobid: 10
    wildcards: sample=B, unit=1</p><p>python /home/alper/tmp/snakemake-dna-seq-gatk-variant-calling-1.0.1/.snakemake/scripts/tmp0ap4algq.wrapper.py
WARNING: The conda.compat module is deprecated and will be removed in a future release.
Activating conda environment: /home/alper/tmp/snakemake-dna-seq-gatk-variant-calling-1.0.1/.snakemake/conda/931cfa38</p><pre spellcheck="false" class="md-fences md-end-block ty-contain-cm modeLoaded" lang=""><div class="CodeMirror cm-s-inner CodeMirror-wrap" lang=""><div style="overflow: hidden; position: relative; width: 3px; height: 0px; top: 0px; left: 4.16669px;"><textarea autocorrect="off" autocapitalize="off" spellcheck="false" tabindex="0" style="position: absolute; bottom: -1em; padding: 0px; width: 1000px; height: 1em; outline: none;"></textarea></div><div class="CodeMirror-scrollbar-filler" cm-not-content="true"></div><div class="CodeMirror-gutter-filler" cm-not-content="true"></div><div class="CodeMirror-scroll" tabindex="-1"><div class="CodeMirror-sizer" style="margin-left: 0px; margin-bottom: 0px; border-right-width: 0px; padding-right: 0px; padding-bottom: 0px;"><div style="position: relative; top: 0px;"><div class="CodeMirror-lines" role="presentation"><div role="presentation" style="position: relative; outline: none;"><div class="CodeMirror-measure"><pre><span>xxxxxxxxxx</span></pre></div><div class="CodeMirror-measure"></div><div style="position: relative; z-index: 1;"></div><div class="CodeMirror-code" role="presentation"><div class="CodeMirror-activeline" style="position: relative;"><div class="CodeMirror-activeline-background CodeMirror-linebackground"></div><div class="CodeMirror-gutter-background CodeMirror-activeline-gutter" style="left: 0px; width: 0px;"></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">picard MarkDuplicates REMOVE_DUPLICATES=true INPUT=mapped/B-1.sorted.bam OUTPUT=dedup/B-1.bam METRICS_FILE=qc/dedup/B-1.metrics.txt &amp;&gt; logs/picard/dedup/B-1.log</span></pre></div></div></div></div></div></div><div style="position: absolute; height: 0px; width: 1px; border-bottom: 0px solid transparent; top: 40px;"></div><div class="CodeMirror-gutters" style="display: none; height: 40px;"></div></div></div></pre><p>Removing temporary output file mapped/B-1.sorted.bam.
[Mon Apr 29 10:37:39 2019]
Finished job 10.
8 of 33 steps (24%) done</p><p>Removing temporary output file mapped/B-1.sorted.bam.
[Mon Apr 29 10:37:39 2019]
Finished job 10.
8 of 33 steps (24%) done</p><p>[Mon Apr 29 10:37:39 2019]
rule recalibrate_base_qualities:
    input: dedup/B-1.bam, data/ref/genome.chr21.fa, data/ref/dbsnp.vcf.gz
    output: recal/B-1.bam
    log: logs/gatk/bqsr/B-1.log
    jobid: 19
    wildcards: sample=B, unit=1</p><p>python /home/alper/tmp/snakemake-dna-seq-gatk-variant-calling-1.0.1/.snakemake/scripts/tmplo6c1ww2.wrapper.py
WARNING: The conda.compat module is deprecated and will be removed in a future release.
Activating conda environment: /home/alper/tmp/snakemake-dna-seq-gatk-variant-calling-1.0.1/.snakemake/conda/21f0a586</p><pre spellcheck="false" class="md-fences md-end-block ty-contain-cm modeLoaded" lang=""><div class="CodeMirror cm-s-inner CodeMirror-wrap" lang=""><div style="overflow: hidden; position: relative; width: 3px; height: 0px; top: 0px; left: 4.16669px;"><textarea autocorrect="off" autocapitalize="off" spellcheck="false" tabindex="0" style="position: absolute; bottom: -1em; padding: 0px; width: 1000px; height: 1em; outline: none;"></textarea></div><div class="CodeMirror-scrollbar-filler" cm-not-content="true"></div><div class="CodeMirror-gutter-filler" cm-not-content="true"></div><div class="CodeMirror-scroll" tabindex="-1"><div class="CodeMirror-sizer" style="margin-left: 0px; margin-bottom: 0px; border-right-width: 0px; padding-right: 0px; padding-bottom: 0px;"><div style="position: relative; top: 0px;"><div class="CodeMirror-lines" role="presentation"><div role="presentation" style="position: relative; outline: none;"><div class="CodeMirror-measure"><pre><span>xxxxxxxxxx</span></pre></div><div class="CodeMirror-measure"></div><div style="position: relative; z-index: 1;"></div><div class="CodeMirror-code" role="presentation"><div class="CodeMirror-activeline" style="position: relative;"><div class="CodeMirror-activeline-background CodeMirror-linebackground"></div><div class="CodeMirror-gutter-background CodeMirror-activeline-gutter" style="left: 0px; width: 0px;"></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">gatk BaseRecalibrator  -R data/ref/genome.chr21.fa -I dedup/B-1.bam -O /tmp/tmpz_7des4i/recal_table.grp --known-sites data/ref/dbsnp.vcf.gz  &gt; logs/gatk/bqsr/B-1.log 2&gt;&amp;1</span></pre></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">gatk ApplyBQSR -R data/ref/genome.chr21.fa -I dedup/B-1.bam --bqsr-recal-file /tmp/tmpz_7des4i/recal_table.grp -O recal/B-1.bam  &gt;&gt; logs/gatk/bqsr/B-1.log 2&gt;&amp;1</span></pre></div></div></div></div></div><div style="position: absolute; height: 0px; width: 1px; border-bottom: 0px solid transparent; top: 100px;"></div><div class="CodeMirror-gutters" style="display: none; height: 100px;"></div></div></div></pre><p>Write-protecting output file recal/B-1.bam.
Removing temporary output file dedup/B-1.bam.
[Mon Apr 29 10:37:47 2019]
Finished job 19.
9 of 33 steps (27%) done</p><p>[Mon Apr 29 10:37:47 2019]
rule map_reads:
    input: trimmed/A-1.1.fastq.gz, trimmed/A-1.2.fastq.gz
    output: mapped/A-1.sorted.bam
    log: logs/bwa_mem/A-1.log
    jobid: 18
    wildcards: sample=A, unit=1
    threads: 8</p><p>python /home/alper/tmp/snakemake-dna-seq-gatk-variant-calling-1.0.1/.snakemake/scripts/tmpiklthufk.wrapper.py
WARNING: The conda.compat module is deprecated and will be removed in a future release.
Activating conda environment: /home/alper/tmp/snakemake-dna-seq-gatk-variant-calling-1.0.1/.snakemake/conda/332db904</p><pre spellcheck="false" class="md-fences md-end-block ty-contain-cm modeLoaded" lang=""><div class="CodeMirror cm-s-inner CodeMirror-wrap" lang=""><div style="overflow: hidden; position: relative; width: 3px; height: 0px; top: 0px; left: 4.16669px;"><textarea autocorrect="off" autocapitalize="off" spellcheck="false" tabindex="0" style="position: absolute; bottom: -1em; padding: 0px; width: 1000px; height: 1em; outline: none;"></textarea></div><div class="CodeMirror-scrollbar-filler" cm-not-content="true"></div><div class="CodeMirror-gutter-filler" cm-not-content="true"></div><div class="CodeMirror-scroll" tabindex="-1"><div class="CodeMirror-sizer" style="margin-left: 0px; margin-bottom: 0px; border-right-width: 0px; padding-right: 0px; padding-bottom: 0px;"><div style="position: relative; top: 0px;"><div class="CodeMirror-lines" role="presentation"><div role="presentation" style="position: relative; outline: none;"><div class="CodeMirror-measure"><pre><span>xxxxxxxxxx</span></pre></div><div class="CodeMirror-measure"></div><div style="position: relative; z-index: 1;"></div><div class="CodeMirror-code" role="presentation"><div class="CodeMirror-activeline" style="position: relative;"><div class="CodeMirror-activeline-background CodeMirror-linebackground"></div><div class="CodeMirror-gutter-background CodeMirror-activeline-gutter" style="left: 0px; width: 0px;"></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">(bwa mem -t 8 -R '@RG\tID:A\tSM:A\tPL:ILLUMINA' data/ref/genome.chr21.fa trimmed/A-1.1.fastq.gz trimmed/A-1.2.fastq.gz | samtools sort  -T mapped/A-1.sorted.tmp -o mapped/A-1.sorted.bam -)  2&gt; logs/bwa_mem/A-1.log</span></pre></div></div></div></div></div></div><div style="position: absolute; height: 0px; width: 1px; border-bottom: 0px solid transparent; top: 60px;"></div><div class="CodeMirror-gutters" style="display: none; height: 60px;"></div></div></div></pre><p>Removing temporary output file trimmed/A-1.1.fastq.gz.
Removing temporary output file trimmed/A-1.2.fastq.gz.
[Mon Apr 29 10:37:50 2019]
Finished job 18.
10 of 33 steps (30%) done</p><p>[Mon Apr 29 10:37:50 2019]
rule mark_duplicates:
    input: mapped/A-1.sorted.bam
    output: dedup/A-1.bam, qc/dedup/A-1.metrics.txt
    log: logs/picard/dedup/A-1.log
    jobid: 7
    wildcards: sample=A, unit=1</p><p>[Mon Apr 29 10:37:50 2019]
rule samtools_stats:
    input: recal/B-1.bam
    output: qc/samtools-stats/B-1.txt
    log: logs/samtools-stats/B-1.log
    jobid: 8
    wildcards: sample=B, unit=1</p><p>python /home/alper/tmp/snakemake-dna-seq-gatk-variant-calling-1.0.1/.snakemake/scripts/tmp80_usdbx.wrapper.py
/home/alper/miniconda3/bin/python3.7 /home/alper/tmp/snakemake-dna-seq-gatk-variant-calling-1.0.1/.snakemake/scripts/tmpyek85neo.wrapper.py
WARNING: The conda.compat module is deprecated and will be removed in a future release.
Activating conda environment: /home/alper/tmp/snakemake-dna-seq-gatk-variant-calling-1.0.1/.snakemake/conda/931cfa38
WARNING: The conda.compat module is deprecated and will be removed in a future release.
Activating conda environment: /home/alper/tmp/snakemake-dna-seq-gatk-variant-calling-1.0.1/.snakemake/conda/c70824eb</p><pre spellcheck="false" class="md-fences md-end-block ty-contain-cm modeLoaded" lang=""><div class="CodeMirror cm-s-inner CodeMirror-wrap" lang=""><div style="overflow: hidden; position: relative; width: 3px; height: 0px; top: 0px; left: 4.16669px;"><textarea autocorrect="off" autocapitalize="off" spellcheck="false" tabindex="0" style="position: absolute; bottom: -1em; padding: 0px; width: 1000px; height: 1em; outline: none;"></textarea></div><div class="CodeMirror-scrollbar-filler" cm-not-content="true"></div><div class="CodeMirror-gutter-filler" cm-not-content="true"></div><div class="CodeMirror-scroll" tabindex="-1"><div class="CodeMirror-sizer" style="margin-left: 0px; margin-bottom: 0px; border-right-width: 0px; padding-right: 0px; padding-bottom: 0px;"><div style="position: relative; top: 0px;"><div class="CodeMirror-lines" role="presentation"><div role="presentation" style="position: relative; outline: none;"><div class="CodeMirror-measure"><pre><span>xxxxxxxxxx</span></pre></div><div class="CodeMirror-measure"></div><div style="position: relative; z-index: 1;"></div><div class="CodeMirror-code" role="presentation"><div class="CodeMirror-activeline" style="position: relative;"><div class="CodeMirror-activeline-background CodeMirror-linebackground"></div><div class="CodeMirror-gutter-background CodeMirror-activeline-gutter" style="left: 0px; width: 0px;"></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">picard MarkDuplicates REMOVE_DUPLICATES=true INPUT=mapped/A-1.sorted.bam OUTPUT=dedup/A-1.bam METRICS_FILE=qc/dedup/A-1.metrics.txt &amp;&gt; logs/picard/dedup/A-1.log</span></pre></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="">​</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">samtools stats  recal/B-1.bam  &gt; qc/samtools-stats/B-1.txt  2&gt; logs/samtools-stats/B-1.log</span></pre></div></div></div></div></div><div style="position: absolute; height: 0px; width: 1px; border-bottom: 0px solid transparent; top: 80px;"></div><div class="CodeMirror-gutters" style="display: none; height: 80px;"></div></div></div></pre><p>[Mon Apr 29 10:37:52 2019]
Finished job 8.
11 of 33 steps (33%) done
Removing temporary output file mapped/A-1.sorted.bam.
[Mon Apr 29 10:37:53 2019]
Finished job 7.
12 of 33 steps (36%) done</p><p>[Mon Apr 29 10:37:53 2019]
rule recalibrate_base_qualities:
    input: dedup/A-1.bam, data/ref/genome.chr21.fa, data/ref/dbsnp.vcf.gz
    output: recal/A-1.bam
    log: logs/gatk/bqsr/A-1.log
    jobid: 17
    wildcards: sample=A, unit=1</p><p>python /home/alper/tmp/snakemake-dna-seq-gatk-variant-calling-1.0.1/.snakemake/scripts/tmpbxlrkm83.wrapper.py
WARNING: The conda.compat module is deprecated and will be removed in a future release.
Activating conda environment: /home/alper/tmp/snakemake-dna-seq-gatk-variant-calling-1.0.1/.snakemake/conda/21f0a586</p><pre spellcheck="false" class="md-fences md-end-block ty-contain-cm modeLoaded" lang=""><div class="CodeMirror cm-s-inner CodeMirror-wrap" lang=""><div style="overflow: hidden; position: relative; width: 3px; height: 0px; top: 0px; left: 4.16669px;"><textarea autocorrect="off" autocapitalize="off" spellcheck="false" tabindex="0" style="position: absolute; bottom: -1em; padding: 0px; width: 1000px; height: 1em; outline: none;"></textarea></div><div class="CodeMirror-scrollbar-filler" cm-not-content="true"></div><div class="CodeMirror-gutter-filler" cm-not-content="true"></div><div class="CodeMirror-scroll" tabindex="-1"><div class="CodeMirror-sizer" style="margin-left: 0px; margin-bottom: 0px; border-right-width: 0px; padding-right: 0px; padding-bottom: 0px;"><div style="position: relative; top: 0px;"><div class="CodeMirror-lines" role="presentation"><div role="presentation" style="position: relative; outline: none;"><div class="CodeMirror-measure"><pre><span>xxxxxxxxxx</span></pre></div><div class="CodeMirror-measure"></div><div style="position: relative; z-index: 1;"></div><div class="CodeMirror-code" role="presentation"><div class="CodeMirror-activeline" style="position: relative;"><div class="CodeMirror-activeline-background CodeMirror-linebackground"></div><div class="CodeMirror-gutter-background CodeMirror-activeline-gutter" style="left: 0px; width: 0px;"></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">gatk BaseRecalibrator  -R data/ref/genome.chr21.fa -I dedup/A-1.bam -O /tmp/tmprrlv1m03/recal_table.grp --known-sites data/ref/dbsnp.vcf.gz  &gt; logs/gatk/bqsr/A-1.log 2&gt;&amp;1</span></pre></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="">​</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">gatk ApplyBQSR -R data/ref/genome.chr21.fa -I dedup/A-1.bam --bqsr-recal-file /tmp/tmprrlv1m03/recal_table.grp -O recal/A-1.bam  &gt;&gt; logs/gatk/bqsr/A-1.log 2&gt;&amp;1</span></pre></div></div></div></div></div><div style="position: absolute; height: 0px; width: 1px; border-bottom: 0px solid transparent; top: 120px;"></div><div class="CodeMirror-gutters" style="display: none; height: 120px;"></div></div></div></pre><blockquote><p>Write-protecting output file recal/A-1.bam.
Removing temporary output file dedup/A-1.bam.</p></blockquote><p>[Mon Apr 29 10:38:01 2019]
Finished job 17.
13 of 33 steps (39%) done</p><p>[Mon Apr 29 10:38:01 2019]
rule map_reads:
    input: trimmed/B-2.fastq.gz
    output: mapped/B-2.sorted.bam
    log: logs/bwa_mem/B-2.log
    jobid: 22
    wildcards: sample=B, unit=2
    threads: 8</p><p>python /home/alper/tmp/snakemake-dna-seq-gatk-variant-calling-1.0.1/.snakemake/scripts/tmpeg4eujj6.wrapper.py
WARNING: The conda.compat module is deprecated and will be removed in a future release.
Activating conda environment: /home/alper/tmp/snakemake-dna-seq-gatk-variant-calling-1.0.1/.snakemake/conda/332db904</p><pre spellcheck="false" class="md-fences md-end-block ty-contain-cm modeLoaded" lang=""><div class="CodeMirror cm-s-inner CodeMirror-wrap" lang=""><div style="overflow: hidden; position: relative; width: 3px; height: 0px; top: 0px; left: 4.16669px;"><textarea autocorrect="off" autocapitalize="off" spellcheck="false" tabindex="0" style="position: absolute; bottom: -1em; padding: 0px; width: 1000px; height: 1em; outline: none;"></textarea></div><div class="CodeMirror-scrollbar-filler" cm-not-content="true"></div><div class="CodeMirror-gutter-filler" cm-not-content="true"></div><div class="CodeMirror-scroll" tabindex="-1"><div class="CodeMirror-sizer" style="margin-left: 0px; margin-bottom: 0px; border-right-width: 0px; padding-right: 0px; padding-bottom: 0px;"><div style="position: relative; top: 0px;"><div class="CodeMirror-lines" role="presentation"><div role="presentation" style="position: relative; outline: none;"><div class="CodeMirror-measure"><pre><span>xxxxxxxxxx</span></pre></div><div class="CodeMirror-measure"></div><div style="position: relative; z-index: 1;"></div><div class="CodeMirror-code" role="presentation"><div class="CodeMirror-activeline" style="position: relative;"><div class="CodeMirror-activeline-background CodeMirror-linebackground"></div><div class="CodeMirror-gutter-background CodeMirror-activeline-gutter" style="left: 0px; width: 0px;"></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">(bwa mem -t 8 -R '@RG\tID:B\tSM:B\tPL:ILLUMINA' data/ref/genome.chr21.fa trimmed/B-2.fastq.gz | samtools sort  -T mapped/B-2.sorted.tmp -o mapped/B-2.sorted.bam -)  2&gt; logs/bwa_mem/B-2.log</span></pre></div></div></div></div></div></div><div style="position: absolute; height: 0px; width: 1px; border-bottom: 0px solid transparent; top: 40px;"></div><div class="CodeMirror-gutters" style="display: none; height: 40px;"></div></div></div></pre><blockquote><p>Removing temporary output file trimmed/B-2.fastq.gz.</p></blockquote><p>[Mon Apr 29 10:38:04 2019]
Finished job 22.
14 of 33 steps (42%) done</p><p>[Mon Apr 29 10:38:04 2019]
rule samtools_stats:
    input: recal/A-1.bam
    output: qc/samtools-stats/A-1.txt
    log: logs/samtools-stats/A-1.log
    jobid: 5
    wildcards: sample=A, unit=1</p><p>[Mon Apr 29 10:38:04 2019]
rule mark_duplicates:
    input: mapped/B-2.sorted.bam
    output: dedup/B-2.bam, qc/dedup/B-2.metrics.txt
    log: logs/picard/dedup/B-2.log
    jobid: 13
    wildcards: sample=B, unit=2</p><p>[Mon Apr 29 10:38:04 2019]
rule call_variants:
    input: recal/A-1.bam, data/ref/genome.chr21.fa, data/ref/dbsnp.vcf.gz
    output: called/A.21.g.vcf.gz
    log: logs/gatk/haplotypecaller/A.21.log
    jobid: 31
    wildcards: sample=A, contig=21</p><p>/home/alper/miniconda3/bin/python3.7 /home/alper/tmp/snakemake-dna-seq-gatk-variant-calling-1.0.1/.snakemake/scripts/tmpsulq8zu5.wrapper.py
python /home/alper/tmp/snakemake-dna-seq-gatk-variant-calling-1.0.1/.snakemake/scripts/tmphcnmlo_b.wrapper.py
python /home/alper/tmp/snakemake-dna-seq-gatk-variant-calling-1.0.1/.snakemake/scripts/tmpdgzoeza1.wrapper.py
WARNING: The conda.compat module is deprecated and will be removed in a future release.
WARNING: The conda.compat module is deprecated and will be removed in a future release.
WARNING: The conda.compat module is deprecated and will be removed in a future release.
Activating conda environment: /home/alper/tmp/snakemake-dna-seq-gatk-variant-calling-1.0.1/.snakemake/conda/931cfa38
Activating conda environment: /home/alper/tmp/snakemake-dna-seq-gatk-variant-calling-1.0.1/.snakemake/conda/21f0a586
Activating conda environment: /home/alper/tmp/snakemake-dna-seq-gatk-variant-calling-1.0.1/.snakemake/conda/c70824eb</p><pre spellcheck="false" class="md-fences md-end-block ty-contain-cm modeLoaded" lang=""><div class="CodeMirror cm-s-inner CodeMirror-wrap" lang=""><div style="overflow: hidden; position: relative; width: 3px; height: 0px; top: 0px; left: 4.16669px;"><textarea autocorrect="off" autocapitalize="off" spellcheck="false" tabindex="0" style="position: absolute; bottom: -1em; padding: 0px; width: 1000px; height: 1em; outline: none;"></textarea></div><div class="CodeMirror-scrollbar-filler" cm-not-content="true"></div><div class="CodeMirror-gutter-filler" cm-not-content="true"></div><div class="CodeMirror-scroll" tabindex="-1"><div class="CodeMirror-sizer" style="margin-left: 0px; margin-bottom: 0px; border-right-width: 0px; padding-right: 0px; padding-bottom: 0px;"><div style="position: relative; top: 0px;"><div class="CodeMirror-lines" role="presentation"><div role="presentation" style="position: relative; outline: none;"><div class="CodeMirror-measure"><pre><span>xxxxxxxxxx</span></pre></div><div class="CodeMirror-measure"></div><div style="position: relative; z-index: 1;"></div><div class="CodeMirror-code" role="presentation" style=""><div class="CodeMirror-activeline" style="position: relative;"><div class="CodeMirror-activeline-background CodeMirror-linebackground"></div><div class="CodeMirror-gutter-background CodeMirror-activeline-gutter" style="left: 0px; width: 0px;"></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">picard MarkDuplicates REMOVE_DUPLICATES=true INPUT=mapped/B-2.sorted.bam OUTPUT=dedup/B-2.bam METRICS_FILE=qc/dedup/B-2.metrics.txt &amp;&gt; logs/picard/dedup/B-2.log</span></pre></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="">​</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">gatk HaplotypeCaller --intervals 21 -R data/ref/genome.chr21.fa -I recal/A-1.bam -ERC GVCF -O called/A.21.g.vcf.gz --dbsnp data/ref/dbsnp.vcf.gz  &gt; logs/gatk/haplotypecaller/A.21.log 2&gt;&amp;1</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="">​</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">samtools stats  recal/A-1.bam  &gt; qc/samtools-stats/A-1.txt  2&gt; logs/samtools-stats/A-1.log</span></pre></div></div></div></div></div><div style="position: absolute; height: 0px; width: 1px; border-bottom: 0px solid transparent; top: 140px;"></div><div class="CodeMirror-gutters" style="display: none; height: 140px;"></div></div></div></pre><p>[Mon Apr 29 10:38:06 2019]
Finished job 5.
15 of 33 steps (45%) done</p><blockquote><p>Removing temporary output file mapped/B-2.sorted.bam.</p></blockquote><p>[Mon Apr 29 10:38:07 2019]
Finished job 13.
16 of 33 steps (48%) done</p><p>[Mon Apr 29 10:38:07 2019]
rule recalibrate_base_qualities:
    input: dedup/B-2.bam, data/ref/genome.chr21.fa, data/ref/dbsnp.vcf.gz
    output: recal/B-2.bam
    log: logs/gatk/bqsr/B-2.log
    jobid: 21
    wildcards: sample=B, unit=2</p><p>python /home/alper/tmp/snakemake-dna-seq-gatk-variant-calling-1.0.1/.snakemake/scripts/tmpb0yjnxhs.wrapper.py
WARNING: The conda.compat module is deprecated and will be removed in a future release.
Activating conda environment: /home/alper/tmp/snakemake-dna-seq-gatk-variant-calling-1.0.1/.snakemake/conda/21f0a586</p><pre spellcheck="false" class="md-fences md-end-block ty-contain-cm modeLoaded" lang=""><div class="CodeMirror cm-s-inner CodeMirror-wrap" lang=""><div style="overflow: hidden; position: relative; width: 3px; height: 0px; top: 0px; left: 4.16669px;"><textarea autocorrect="off" autocapitalize="off" spellcheck="false" tabindex="0" style="position: absolute; bottom: -1em; padding: 0px; width: 1000px; height: 1em; outline: none;"></textarea></div><div class="CodeMirror-scrollbar-filler" cm-not-content="true"></div><div class="CodeMirror-gutter-filler" cm-not-content="true"></div><div class="CodeMirror-scroll" tabindex="-1"><div class="CodeMirror-sizer" style="margin-left: 0px; margin-bottom: 0px; border-right-width: 0px; padding-right: 0px; padding-bottom: 0px;"><div style="position: relative; top: 0px;"><div class="CodeMirror-lines" role="presentation"><div role="presentation" style="position: relative; outline: none;"><div class="CodeMirror-measure"><pre><span>xxxxxxxxxx</span></pre></div><div class="CodeMirror-measure"></div><div style="position: relative; z-index: 1;"></div><div class="CodeMirror-code" role="presentation"><div class="CodeMirror-activeline" style="position: relative;"><div class="CodeMirror-activeline-background CodeMirror-linebackground"></div><div class="CodeMirror-gutter-background CodeMirror-activeline-gutter" style="left: 0px; width: 0px;"></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">gatk BaseRecalibrator  -R data/ref/genome.chr21.fa -I dedup/B-2.bam -O /tmp/tmpis2ryhs1/recal_table.grp --known-sites data/ref/dbsnp.vcf.gz  &gt; logs/gatk/bqsr/B-2.log 2&gt;&amp;1</span></pre></div><div class="" style="position: relative;"><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="">​</span></span></pre></div><div class="" style="position: relative;"><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">gatk ApplyBQSR -R data/ref/genome.chr21.fa -I dedup/B-2.bam --bqsr-recal-file /tmp/tmpis2ryhs1/recal_table.grp -O recal/B-2.bam  &gt;&gt; logs/gatk/bqsr/B-2.log 2&gt;&amp;1</span></pre></div></div></div></div></div></div><div style="position: absolute; height: 0px; width: 1px; border-bottom: 0px solid transparent; top: 120px;"></div><div class="CodeMirror-gutters" style="display: none; height: 120px;"></div></div></div></pre><blockquote><p>Write-protecting output file recal/B-2.bam.
Removing temporary output file dedup/B-2.bam.</p></blockquote><p>[Mon Apr 29 10:38:17 2019]
Finished job 21.
17 of 33 steps (52%) done</p><p>[Mon Apr 29 10:38:17 2019]
rule samtools_stats:
    input: recal/B-2.bam
    output: qc/samtools-stats/B-2.txt
    log: logs/samtools-stats/B-2.log
    jobid: 11
    wildcards: sample=B, unit=2</p><p>[Mon Apr 29 10:38:17 2019]
rule call_variants:
    input: recal/B-1.bam, recal/B-2.bam, data/ref/genome.chr21.fa, data/ref/dbsnp.vcf.gz
    output: called/B.21.g.vcf.gz
    log: logs/gatk/haplotypecaller/B.21.log
    jobid: 32
    wildcards: sample=B, contig=21</p><p>python /home/alper/tmp/snakemake-dna-seq-gatk-variant-calling-1.0.1/.snakemake/scripts/tmptu4i6edy.wrapper.py
/home/alper/miniconda3/bin/python3.7 /home/alper/tmp/snakemake-dna-seq-gatk-variant-calling-1.0.1/.snakemake/scripts/tmpmhcgf_1k.wrapper.py
WARNING: The conda.compat module is deprecated and will be removed in a future release.
WARNING: The conda.compat module is deprecated and will be removed in a future release.
Activating conda environment: /home/alper/tmp/snakemake-dna-seq-gatk-variant-calling-1.0.1/.snakemake/conda/c70824eb
Activating conda environment: /home/alper/tmp/snakemake-dna-seq-gatk-variant-calling-1.0.1/.snakemake/conda/21f0a586</p><pre spellcheck="false" class="md-fences md-end-block ty-contain-cm modeLoaded" lang=""><div class="CodeMirror cm-s-inner CodeMirror-wrap" lang=""><div style="overflow: hidden; position: relative; width: 3px; height: 0px; top: 0px; left: 4.16669px;"><textarea autocorrect="off" autocapitalize="off" spellcheck="false" tabindex="0" style="position: absolute; bottom: -1em; padding: 0px; width: 1000px; height: 1em; outline: none;"></textarea></div><div class="CodeMirror-scrollbar-filler" cm-not-content="true"></div><div class="CodeMirror-gutter-filler" cm-not-content="true"></div><div class="CodeMirror-scroll" tabindex="-1"><div class="CodeMirror-sizer" style="margin-left: 0px; margin-bottom: 0px; border-right-width: 0px; padding-right: 0px; padding-bottom: 0px;"><div style="position: relative; top: 0px;"><div class="CodeMirror-lines" role="presentation"><div role="presentation" style="position: relative; outline: none;"><div class="CodeMirror-measure"><pre><span>xxxxxxxxxx</span></pre></div><div class="CodeMirror-measure"></div><div style="position: relative; z-index: 1;"></div><div class="CodeMirror-code" role="presentation"><div class="CodeMirror-activeline" style="position: relative;"><div class="CodeMirror-activeline-background CodeMirror-linebackground"></div><div class="CodeMirror-gutter-background CodeMirror-activeline-gutter" style="left: 0px; width: 0px;"></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">gatk HaplotypeCaller --intervals 21 -R data/ref/genome.chr21.fa -I recal/B-1.bam -I recal/B-2.bam -ERC GVCF -O called/B.21.g.vcf.gz --dbsnp data/ref/dbsnp.vcf.gz  &gt; logs/gatk/haplotypecaller/B.21.log 2&gt;&amp;1</span></pre></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="">​</span></span></pre><div class="" style="position: relative;"><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">samtools stats  recal/B-2.bam  &gt; qc/samtools-stats/B-2.txt  2&gt; logs/samtools-stats/B-2.log</span></pre></div></div></div></div></div></div><div style="position: absolute; height: 0px; width: 1px; border-bottom: 0px solid transparent; top: 100px;"></div><div class="CodeMirror-gutters" style="display: none; height: 100px;"></div></div></div></pre><p>[Mon Apr 29 10:38:19 2019]
Finished job 11.
18 of 33 steps (55%) done</p><blockquote><p>Write-protecting output file called/A.21.g.vcf.gz.</p></blockquote><p>[Mon Apr 29 10:39:05 2019]
Finished job 31.
19 of 33 steps (58%) done</p><blockquote><p>Write-protecting output file called/B.21.g.vcf.gz.</p></blockquote><p>[Mon Apr 29 10:39:19 2019]
Finished job 32.
20 of 33 steps (61%) done</p><p>[Mon Apr 29 10:39:19 2019]
rule combine_calls:
    input: data/ref/genome.chr21.fa, called/A.21.g.vcf.gz, called/B.21.g.vcf.gz
    output: called/all.21.g.vcf.gz
    log: logs/gatk/combinegvcfs.21.log
    jobid: 30
    wildcards: contig=21</p><p>python /home/alper/tmp/snakemake-dna-seq-gatk-variant-calling-1.0.1/.snakemake/scripts/tmpxf7pu2_4.wrapper.py
WARNING: The conda.compat module is deprecated and will be removed in a future release.
Activating conda environment: /home/alper/tmp/snakemake-dna-seq-gatk-variant-calling-1.0.1/.snakemake/conda/21f0a586</p><pre spellcheck="false" class="md-fences md-end-block ty-contain-cm modeLoaded" lang=""><div class="CodeMirror cm-s-inner CodeMirror-wrap" lang=""><div style="overflow: hidden; position: relative; width: 3px; height: 0px; top: 0px; left: 4.16669px;"><textarea autocorrect="off" autocapitalize="off" spellcheck="false" tabindex="0" style="position: absolute; bottom: -1em; padding: 0px; width: 1000px; height: 1em; outline: none;"></textarea></div><div class="CodeMirror-scrollbar-filler" cm-not-content="true"></div><div class="CodeMirror-gutter-filler" cm-not-content="true"></div><div class="CodeMirror-scroll" tabindex="-1"><div class="CodeMirror-sizer" style="margin-left: 0px; margin-bottom: 0px; border-right-width: 0px; padding-right: 0px; padding-bottom: 0px;"><div style="position: relative; top: 0px;"><div class="CodeMirror-lines" role="presentation"><div role="presentation" style="position: relative; outline: none;"><div class="CodeMirror-measure"><pre><span>xxxxxxxxxx</span></pre></div><div class="CodeMirror-measure"></div><div style="position: relative; z-index: 1;"></div><div class="CodeMirror-code" role="presentation"><div class="CodeMirror-activeline" style="position: relative;"><div class="CodeMirror-activeline-background CodeMirror-linebackground"></div><div class="CodeMirror-gutter-background CodeMirror-activeline-gutter" style="left: 0px; width: 0px;"></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">gatk CombineGVCFs  -V called/A.21.g.vcf.gz -V called/B.21.g.vcf.gz -R data/ref/genome.chr21.fa -O called/all.21.g.vcf.gz  &gt; logs/gatk/combinegvcfs.21.log 2&gt;&amp;1</span></pre></div></div></div></div></div></div><div style="position: absolute; height: 0px; width: 1px; border-bottom: 0px solid transparent; top: 40px;"></div><div class="CodeMirror-gutters" style="display: none; height: 40px;"></div></div></div></pre><p>[Mon Apr 29 10:39:26 2019]
Finished job 30.
21 of 33 steps (64%) done</p><p>[Mon Apr 29 10:39:26 2019]
rule genotype_variants:
    input: data/ref/genome.chr21.fa, called/all.21.g.vcf.gz
    output: genotyped/all.21.vcf.gz
    log: logs/gatk/genotypegvcfs.21.log
    jobid: 29
    wildcards: contig=21</p><p>python /home/alper/tmp/snakemake-dna-seq-gatk-variant-calling-1.0.1/.snakemake/scripts/tmpwchpfpyy.wrapper.py
WARNING: The conda.compat module is deprecated and will be removed in a future release.
Activating conda environment: /home/alper/tmp/snakemake-dna-seq-gatk-variant-calling-1.0.1/.snakemake/conda/21f0a586</p><pre spellcheck="false" class="md-fences md-end-block ty-contain-cm modeLoaded" lang=""><div class="CodeMirror cm-s-inner CodeMirror-wrap" lang=""><div style="overflow: hidden; position: relative; width: 3px; height: 0px; top: 0px; left: 4.16669px;"><textarea autocorrect="off" autocapitalize="off" spellcheck="false" tabindex="0" style="position: absolute; bottom: -1em; padding: 0px; width: 1000px; height: 1em; outline: none;"></textarea></div><div class="CodeMirror-scrollbar-filler" cm-not-content="true"></div><div class="CodeMirror-gutter-filler" cm-not-content="true"></div><div class="CodeMirror-scroll" tabindex="-1"><div class="CodeMirror-sizer" style="margin-left: 0px; margin-bottom: 0px; border-right-width: 0px; padding-right: 0px; padding-bottom: 0px;"><div style="position: relative; top: 0px;"><div class="CodeMirror-lines" role="presentation"><div role="presentation" style="position: relative; outline: none;"><div class="CodeMirror-measure"><pre><span>xxxxxxxxxx</span></pre></div><div class="CodeMirror-measure"></div><div style="position: relative; z-index: 1;"></div><div class="CodeMirror-code" role="presentation"><div class="CodeMirror-activeline" style="position: relative;"><div class="CodeMirror-activeline-background CodeMirror-linebackground"></div><div class="CodeMirror-gutter-background CodeMirror-activeline-gutter" style="left: 0px; width: 0px;"></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">gatk GenotypeGVCFs  -V called/all.21.g.vcf.gz -R data/ref/genome.chr21.fa -O genotyped/all.21.vcf.gz  &gt; logs/gatk/genotypegvcfs.21.log 2&gt;&amp;1</span></pre></div></div></div></div></div></div><div style="position: absolute; height: 0px; width: 1px; border-bottom: 0px solid transparent; top: 40px;"></div><div class="CodeMirror-gutters" style="display: none; height: 40px;"></div></div></div></pre><p>[Mon Apr 29 10:39:32 2019]
Finished job 29.
22 of 33 steps (67%) done</p><p>[Mon Apr 29 10:39:32 2019]
rule merge_variants:
    input: genotyped/all.21.vcf.gz
    output: genotyped/all.vcf.gz
    log: logs/picard/merge-genotyped.log
    jobid: 28</p><p>python /home/alper/tmp/snakemake-dna-seq-gatk-variant-calling-1.0.1/.snakemake/scripts/tmpklx955xr.wrapper.py
WARNING: The conda.compat module is deprecated and will be removed in a future release.
Activating conda environment: /home/alper/tmp/snakemake-dna-seq-gatk-variant-calling-1.0.1/.snakemake/conda/931cfa38</p><pre spellcheck="false" class="md-fences md-end-block ty-contain-cm modeLoaded" lang=""><div class="CodeMirror cm-s-inner CodeMirror-wrap" lang=""><div style="overflow: hidden; position: relative; width: 3px; height: 0px; top: 0px; left: 4.16669px;"><textarea autocorrect="off" autocapitalize="off" spellcheck="false" tabindex="0" style="position: absolute; bottom: -1em; padding: 0px; width: 1000px; height: 1em; outline: none;"></textarea></div><div class="CodeMirror-scrollbar-filler" cm-not-content="true"></div><div class="CodeMirror-gutter-filler" cm-not-content="true"></div><div class="CodeMirror-scroll" tabindex="-1"><div class="CodeMirror-sizer" style="margin-left: 0px; margin-bottom: 0px; border-right-width: 0px; padding-right: 0px; padding-bottom: 0px;"><div style="position: relative; top: 0px;"><div class="CodeMirror-lines" role="presentation"><div role="presentation" style="position: relative; outline: none;"><div class="CodeMirror-measure"><pre><span>xxxxxxxxxx</span></pre></div><div class="CodeMirror-measure"></div><div style="position: relative; z-index: 1;"></div><div class="CodeMirror-code" role="presentation"><div class="CodeMirror-activeline" style="position: relative;"><div class="CodeMirror-activeline-background CodeMirror-linebackground"></div><div class="CodeMirror-gutter-background CodeMirror-activeline-gutter" style="left: 0px; width: 0px;"></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">picard MergeVcfs  INPUT=genotyped/all.21.vcf.gz OUTPUT=genotyped/all.vcf.gz  2&gt; logs/picard/merge-genotyped.log</span></pre></div></div></div></div></div></div><div style="position: absolute; height: 0px; width: 1px; border-bottom: 0px solid transparent; top: 40px;"></div><div class="CodeMirror-gutters" style="display: none; height: 40px;"></div></div></div></pre><blockquote><p>Removing temporary output file genotyped/all.21.vcf.gz.</p></blockquote><p>[Mon Apr 29 10:39:35 2019]
Finished job 28.
23 of 33 steps (70%) done</p><p>[Mon Apr 29 10:39:35 2019]
rule select_calls:
    input: data/ref/genome.chr21.fa, genotyped/all.vcf.gz
    output: filtered/all.snvs.vcf.gz
    log: logs/gatk/selectvariants/snvs.log
    jobid: 23
    wildcards: vartype=snvs</p><p>[Mon Apr 29 10:39:35 2019]
rule select_calls:
    input: data/ref/genome.chr21.fa, genotyped/all.vcf.gz
    output: filtered/all.indels.vcf.gz
    log: logs/gatk/selectvariants/indels.log
    jobid: 24
    wildcards: vartype=indels</p><p>python /home/alper/tmp/snakemake-dna-seq-gatk-variant-calling-1.0.1/.snakemake/scripts/tmpscualt3h.wrapper.py
python /home/alper/tmp/snakemake-dna-seq-gatk-variant-calling-1.0.1/.snakemake/scripts/tmpl6rfir0b.wrapper.py
WARNING: The conda.compat module is deprecated and will be removed in a future release.
WARNING: The conda.compat module is deprecated and will be removed in a future release.
Activating conda environment: /home/alper/tmp/snakemake-dna-seq-gatk-variant-calling-1.0.1/.snakemake/conda/21f0a586
Activating conda environment: /home/alper/tmp/snakemake-dna-seq-gatk-variant-calling-1.0.1/.snakemake/conda/21f0a586</p><pre spellcheck="false" class="md-fences md-end-block ty-contain-cm modeLoaded" lang=""><div class="CodeMirror cm-s-inner CodeMirror-wrap" lang=""><div style="overflow: hidden; position: relative; width: 3px; height: 0px; top: 0px; left: 4.16669px;"><textarea autocorrect="off" autocapitalize="off" spellcheck="false" tabindex="0" style="position: absolute; bottom: -1em; padding: 0px; width: 1000px; height: 1em; outline: none;"></textarea></div><div class="CodeMirror-scrollbar-filler" cm-not-content="true"></div><div class="CodeMirror-gutter-filler" cm-not-content="true"></div><div class="CodeMirror-scroll" tabindex="-1"><div class="CodeMirror-sizer" style="margin-left: 0px; margin-bottom: 0px; border-right-width: 0px; padding-right: 0px; padding-bottom: 0px;"><div style="position: relative; top: 0px;"><div class="CodeMirror-lines" role="presentation"><div role="presentation" style="position: relative; outline: none;"><div class="CodeMirror-measure"><pre><span>xxxxxxxxxx</span></pre></div><div class="CodeMirror-measure"></div><div style="position: relative; z-index: 1;"></div><div class="CodeMirror-code" role="presentation"><div class="CodeMirror-activeline" style="position: relative;"><div class="CodeMirror-activeline-background CodeMirror-linebackground"></div><div class="CodeMirror-gutter-background CodeMirror-activeline-gutter" style="left: 0px; width: 0px;"></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">gatk SelectVariants -R data/ref/genome.chr21.fa -V genotyped/all.vcf.gz --select-type-to-include SNP -O filtered/all.snvs.vcf.gz  &gt; logs/gatk/selectvariants/snvs.log 2&gt;&amp;1</span></pre></div><div class="" style="position: relative;"><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="">​</span></span></pre></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">gatk SelectVariants -R data/ref/genome.chr21.fa -V genotyped/all.vcf.gz --select-type-to-include INDEL -O filtered/all.indels.vcf.gz  &gt; logs/gatk/selectvariants/indels.log 2&gt;&amp;1</span></pre></div></div></div></div></div><div style="position: absolute; height: 0px; width: 1px; border-bottom: 0px solid transparent; top: 100px;"></div><div class="CodeMirror-gutters" style="display: none; height: 100px;"></div></div></div></pre><p>[Mon Apr 29 10:39:40 2019]
Finished job 24.
24 of 33 steps (73%) done</p><p>[Mon Apr 29 10:39:40 2019]
rule hard_filter_calls:
    input: data/ref/genome.chr21.fa, filtered/all.indels.vcf.gz
    output: filtered/all.indels.hardfiltered.vcf.gz
    log: logs/gatk/variantfiltration/indels.log
    jobid: 16
    wildcards: vartype=indels</p><p>[Mon Apr 29 10:39:40 2019]
Finished job 23.
25 of 33 steps (76%) done</p><p>[Mon Apr 29 10:39:40 2019]
rule hard_filter_calls:
    input: data/ref/genome.chr21.fa, filtered/all.snvs.vcf.gz
    output: filtered/all.snvs.hardfiltered.vcf.gz
    log: logs/gatk/variantfiltration/snvs.log
    jobid: 15
    wildcards: vartype=snvs</p><p>python /home/alper/tmp/snakemake-dna-seq-gatk-variant-calling-1.0.1/.snakemake/scripts/tmp0gwimrlg.wrapper.py
python /home/alper/tmp/snakemake-dna-seq-gatk-variant-calling-1.0.1/.snakemake/scripts/tmp335528m0.wrapper.py
WARNING: The conda.compat module is deprecated and will be removed in a future release.
Activating conda environment: /home/alper/tmp/snakemake-dna-seq-gatk-variant-calling-1.0.1/.snakemake/conda/21f0a586
WARNING: The conda.compat module is deprecated and will be removed in a future release.
Activating conda environment: /home/alper/tmp/snakemake-dna-seq-gatk-variant-calling-1.0.1/.snakemake/conda/21f0a586</p><pre spellcheck="false" class="md-fences md-end-block ty-contain-cm modeLoaded" lang=""><div class="CodeMirror cm-s-inner CodeMirror-wrap" lang=""><div style="overflow: hidden; position: relative; width: 3px; height: 0px; top: 0px; left: 4.16669px;"><textarea autocorrect="off" autocapitalize="off" spellcheck="false" tabindex="0" style="position: absolute; bottom: -1em; padding: 0px; width: 1000px; height: 1em; outline: none;"></textarea></div><div class="CodeMirror-scrollbar-filler" cm-not-content="true"></div><div class="CodeMirror-gutter-filler" cm-not-content="true"></div><div class="CodeMirror-scroll" tabindex="-1"><div class="CodeMirror-sizer" style="margin-left: 0px; margin-bottom: 0px; border-right-width: 0px; padding-right: 0px; padding-bottom: 0px;"><div style="position: relative; top: 0px;"><div class="CodeMirror-lines" role="presentation"><div role="presentation" style="position: relative; outline: none;"><div class="CodeMirror-measure"><pre><span>xxxxxxxxxx</span></pre></div><div class="CodeMirror-measure"></div><div style="position: relative; z-index: 1;"></div><div class="CodeMirror-code" role="presentation"><div class="CodeMirror-activeline" style="position: relative;"><div class="CodeMirror-activeline-background CodeMirror-linebackground"></div><div class="CodeMirror-gutter-background CodeMirror-activeline-gutter" style="left: 0px; width: 0px;"></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">gatk VariantFiltration -R data/ref/genome.chr21.fa -V filtered/all.indels.vcf.gz  --filter-name snv-hard-filter --filter-expression 'QD &lt; 2.0 || FS &gt; 200.0 || ReadPosRankSum &lt; -20.0' -O filtered/all.indels.hardfiltered.vcf.gz  &gt; logs/gatk/variantfiltration/indels.log 2&gt;&amp;1</span></pre></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="">​</span></span></pre><div class="" style="position: relative;"><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">gatk VariantFiltration -R data/ref/genome.chr21.fa -V filtered/all.snvs.vcf.gz  --filter-name snv-hard-filter --filter-expression 'QD &lt; 2.0 || FS &gt; 60.0 || MQ &lt; 40.0 || MQRankSum &lt; -12.5 || ReadPosRankSum &lt; -8.0' -O filtered/all.snvs.hardfiltered.vcf.gz  &gt; logs/gatk/variantfiltration/snvs.log 2&gt;&amp;1</span></pre></div></div></div></div></div></div><div style="position: absolute; height: 0px; width: 1px; border-bottom: 0px solid transparent; top: 160px;"></div><div class="CodeMirror-gutters" style="display: none; height: 160px;"></div></div></div></pre><blockquote><p>Removing temporary output file filtered/all.indels.vcf.gz.</p></blockquote><p>[Mon Apr 29 10:39:45 2019]
Finished job 16.
26 of 33 steps (79%) done</p><blockquote><p>Removing temporary output file filtered/all.snvs.vcf.gz.</p></blockquote><p>[Mon Apr 29 10:39:45 2019]
Finished job 15.
27 of 33 steps (82%) done</p><p>[Mon Apr 29 10:39:45 2019]
rule merge_calls:
    input: filtered/all.snvs.hardfiltered.vcf.gz, filtered/all.indels.hardfiltered.vcf.gz
    output: filtered/all.vcf.gz
    log: logs/picard/merge-filtered.log
    jobid: 4</p><p>python /home/alper/tmp/snakemake-dna-seq-gatk-variant-calling-1.0.1/.snakemake/scripts/tmpc6ke25oy.wrapper.py
WARNING: The conda.compat module is deprecated and will be removed in a future release.
Activating conda environment: /home/alper/tmp/snakemake-dna-seq-gatk-variant-calling-1.0.1/.snakemake/conda/931cfa38</p><pre spellcheck="false" class="md-fences md-end-block ty-contain-cm modeLoaded" lang=""><div class="CodeMirror cm-s-inner CodeMirror-wrap" lang=""><div style="overflow: hidden; position: relative; width: 3px; height: 0px; top: 0px; left: 4.16669px;"><textarea autocorrect="off" autocapitalize="off" spellcheck="false" tabindex="0" style="position: absolute; bottom: -1em; padding: 0px; width: 1000px; height: 1em; outline: none;"></textarea></div><div class="CodeMirror-scrollbar-filler" cm-not-content="true"></div><div class="CodeMirror-gutter-filler" cm-not-content="true"></div><div class="CodeMirror-scroll" tabindex="-1"><div class="CodeMirror-sizer" style="margin-left: 0px; margin-bottom: 0px; border-right-width: 0px; padding-right: 0px; padding-bottom: 0px;"><div style="position: relative; top: 0px;"><div class="CodeMirror-lines" role="presentation"><div role="presentation" style="position: relative; outline: none;"><div class="CodeMirror-measure"><pre><span>xxxxxxxxxx</span></pre></div><div class="CodeMirror-measure"></div><div style="position: relative; z-index: 1;"></div><div class="CodeMirror-code" role="presentation"><div class="CodeMirror-activeline" style="position: relative;"><div class="CodeMirror-activeline-background CodeMirror-linebackground"></div><div class="CodeMirror-gutter-background CodeMirror-activeline-gutter" style="left: 0px; width: 0px;"></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">picard MergeVcfs  INPUT=filtered/all.snvs.hardfiltered.vcf.gz INPUT=filtered/all.indels.hardfiltered.vcf.gz OUTPUT=filtered/all.vcf.gz  2&gt; logs/picard/merge-filtered.log</span></pre></div></div></div></div></div></div><div style="position: absolute; height: 0px; width: 1px; border-bottom: 0px solid transparent; top: 60px;"></div><div class="CodeMirror-gutters" style="display: none; height: 60px;"></div></div></div></pre><blockquote><p>Removing temporary output file filtered/all.snvs.hardfiltered.vcf.gz.
Removing temporary output file filtered/all.indels.hardfiltered.vcf.gz.</p></blockquote><p>[Mon Apr 29 10:39:48 2019]
Finished job 4.
28 of 33 steps (85%) done</p><p>[Mon Apr 29 10:39:48 2019]
rule snpeff:
    input: filtered/all.vcf.gz
    output: annotated/all.vcf.gz, snpeff/all.csv
    log: logs/snpeff.log
    jobid: 1</p><p>/home/alper/miniconda3/bin/python3.7 /home/alper/tmp/snakemake-dna-seq-gatk-variant-calling-1.0.1/.snakemake/scripts/tmpizt_itqb.wrapper.py
WARNING: The conda.compat module is deprecated and will be removed in a future release.
Activating conda environment: /home/alper/tmp/snakemake-dna-seq-gatk-variant-calling-1.0.1/.snakemake/conda/9a672934</p><pre spellcheck="false" class="md-fences md-end-block ty-contain-cm modeLoaded" lang=""><div class="CodeMirror cm-s-inner CodeMirror-wrap" lang=""><div style="overflow: hidden; position: relative; width: 3px; height: 0px; top: 0px; left: 4.16669px;"><textarea autocorrect="off" autocapitalize="off" spellcheck="false" tabindex="0" style="position: absolute; bottom: -1em; padding: 0px; width: 1000px; height: 1em; outline: none;"></textarea></div><div class="CodeMirror-scrollbar-filler" cm-not-content="true"></div><div class="CodeMirror-gutter-filler" cm-not-content="true"></div><div class="CodeMirror-scroll" tabindex="-1"><div class="CodeMirror-sizer" style="margin-left: 0px; margin-bottom: 0px; border-right-width: 0px; padding-right: 0px; padding-bottom: 0px;"><div style="position: relative; top: 0px;"><div class="CodeMirror-lines" role="presentation"><div role="presentation" style="position: relative; outline: none;"><div class="CodeMirror-measure"><pre><span>xxxxxxxxxx</span></pre></div><div class="CodeMirror-measure"></div><div style="position: relative; z-index: 1;"></div><div class="CodeMirror-code" role="presentation"><div class="CodeMirror-activeline" style="position: relative;"><div class="CodeMirror-activeline-background CodeMirror-linebackground"></div><div class="CodeMirror-gutter-background CodeMirror-activeline-gutter" style="left: 0px; width: 0px;"></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">(snpEff  -noStats -csvStats snpeff/all.csv -Xmx6g GRCh38.86 filtered/all.vcf.gz &gt; annotated/all.vcf.gz)  2&gt; logs/snpeff.log</span></pre></div></div></div></div></div></div><div style="position: absolute; height: 0px; width: 1px; border-bottom: 0px solid transparent; top: 40px;"></div><div class="CodeMirror-gutters" style="display: none; height: 40px;"></div></div></div></pre><p>[Mon Apr 29 10:40:23 2019]
Finished job 1.
29 of 33 steps (88%) done</p><p>[Mon Apr 29 10:40:23 2019]
rule vcf_to_tsv:
    input: annotated/all.vcf.gz
    output: tables/calls.tsv.gz
    jobid: 14</p><pre spellcheck="false" class="md-fences md-end-block ty-contain-cm modeLoaded" lang=""><div class="CodeMirror cm-s-inner CodeMirror-wrap" lang=""><div style="overflow: hidden; position: relative; width: 3px; height: 0px; top: 0px; left: 4.16669px;"><textarea autocorrect="off" autocapitalize="off" spellcheck="false" tabindex="0" style="position: absolute; bottom: -1em; padding: 0px; width: 1000px; height: 1em; outline: none;"></textarea></div><div class="CodeMirror-scrollbar-filler" cm-not-content="true"></div><div class="CodeMirror-gutter-filler" cm-not-content="true"></div><div class="CodeMirror-scroll" tabindex="-1"><div class="CodeMirror-sizer" style="margin-left: 0px; margin-bottom: 0px; border-right-width: 0px; padding-right: 0px; padding-bottom: 0px;"><div style="position: relative; top: 0px;"><div class="CodeMirror-lines" role="presentation"><div role="presentation" style="position: relative; outline: none;"><div class="CodeMirror-measure"><pre><span>xxxxxxxxxx</span></pre></div><div class="CodeMirror-measure"></div><div style="position: relative; z-index: 1;"></div><div class="CodeMirror-code" role="presentation"><div class="CodeMirror-activeline" style="position: relative;"><div class="CodeMirror-activeline-background CodeMirror-linebackground"></div><div class="CodeMirror-gutter-background CodeMirror-activeline-gutter" style="left: 0px; width: 0px;"></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">bcftools view --apply-filters PASS --output-type u annotated/all.vcf.gz | rbt vcf-to-txt -g --fmt DP AD --info ANN | gzip &gt; tables/calls.tsv.gz</span></pre></div></div></div></div></div></div><div style="position: absolute; height: 0px; width: 1px; border-bottom: 0px solid transparent; top: 40px;"></div><div class="CodeMirror-gutters" style="display: none; height: 40px;"></div></div></div></pre><p>[Mon Apr 29 10:40:23 2019]
rule multiqc:
    input: qc/samtools-stats/A-1.txt, qc/fastqc/A-1.zip, qc/dedup/A-1.metrics.txt, qc/samtools-stats/B-1.txt, qc/fastqc/B-1.zip, qc/dedup/B-1.metrics.txt, qc/samtools-stats/B-2.txt, qc/fastqc/B-2.zip, qc/dedup/B-2.metrics.txt, snpeff/all.csv
    output: qc/multiqc.html
    log: logs/multiqc.log
    jobid: 2</p><p>WARNING: The conda.compat module is deprecated and will be removed in a future release.
Activating conda environment: /home/alper/tmp/snakemake-dna-seq-gatk-variant-calling-1.0.1/.snakemake/conda/783643b9
[Mon Apr 29 10:40:24 2019]
Finished job 14.
30 of 33 steps (91%) done</p><p>[Mon Apr 29 10:40:24 2019]
rule plot_stats:
    input: tables/calls.tsv.gz
    output: plots/depths.svg, plots/allele-freqs.svg
    jobid: 3</p><p>python /home/alper/tmp/snakemake-dna-seq-gatk-variant-calling-1.0.1/.snakemake/scripts/tmpbp1vlae6.plot-depths.py
WARNING: The conda.compat module is deprecated and will be removed in a future release.
Activating conda environment: /home/alper/tmp/snakemake-dna-seq-gatk-variant-calling-1.0.1/.snakemake/conda/4c3018c1
Conda environment defines Python version &lt; 3.5. Using Python of the master process to execute script. Note that this cannot be avoided, because the script uses data structures from Snakemake which are Python &gt;=3.5 only.
/home/alper/miniconda3/bin/python3.7 /home/alper/tmp/snakemake-dna-seq-gatk-variant-calling-1.0.1/.snakemake/scripts/tmp4ctdm_gu.wrapper.py
WARNING: The conda.compat module is deprecated and will be removed in a future release.
Activating conda environment: /home/alper/tmp/snakemake-dna-seq-gatk-variant-calling-1.0.1/.snakemake/conda/b822cd76
/home/alper/tmp/snakemake-dna-seq-gatk-variant-calling-1.0.1/scripts/plot-depths.py:11: FutureWarning: read_table is deprecated, use read_csv instead, passing sep=&#39;\t&#39;.</p><pre spellcheck="false" class="md-fences md-end-block ty-contain-cm modeLoaded md-focus" lang=""><div class="CodeMirror cm-s-inner CodeMirror-wrap CodeMirror-focused" lang=""><div style="overflow: hidden; position: relative; width: 3px; height: 0px; top: 19.5835px; left: 173.333px;"><textarea autocorrect="off" autocapitalize="off" spellcheck="false" tabindex="0" style="position: absolute; bottom: -1em; padding: 0px; width: 1000px; height: 1em; outline: none;"></textarea></div><div class="CodeMirror-scrollbar-filler" cm-not-content="true"></div><div class="CodeMirror-gutter-filler" cm-not-content="true"></div><div class="CodeMirror-scroll" tabindex="-1"><div class="CodeMirror-sizer" style="margin-left: 0px; margin-bottom: 0px; border-right-width: 0px; padding-right: 0px; padding-bottom: 0px;"><div style="position: relative; top: 0px;"><div class="CodeMirror-lines" role="presentation"><div role="presentation" style="position: relative; outline: none;"><div class="CodeMirror-measure"><pre><span>xxxxxxxxxx</span></pre></div><div class="CodeMirror-measure"></div><div style="position: relative; z-index: 1;"></div><div class="CodeMirror-code" role="presentation"><div class="CodeMirror-activeline" style="position: relative;"><div class="CodeMirror-activeline-background CodeMirror-linebackground"></div><div class="CodeMirror-gutter-background CodeMirror-activeline-gutter" style="left: 0px; width: 0px;"></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">multiqc  --force -o qc -n multiqc.html qc/samtools-stats qc/dedup snpeff qc/fastqc  &gt; logs/multiqc.log 2&gt;&amp;1</span></pre></div></div></div></div></div></div><div style="position: absolute; height: 0px; width: 1px; border-bottom: 0px solid transparent; top: 40px;"></div><div class="CodeMirror-gutters" style="display: none; height: 40px;"></div></div></div></pre><p>[Mon Apr 29 10:40:25 2019]
Finished job 3.
31 of 33 steps (94%) done
[Mon Apr 29 10:40:28 2019]
Finished job 2.
32 of 33 steps (97%) done</p><p>[Mon Apr 29 10:40:28 2019]
localrule all:
    input: annotated/all.vcf.gz, qc/multiqc.html, plots/depths.svg, plots/allele-freqs.svg
    jobid: 0</p><p>[Mon Apr 29 10:40:28 2019]
Finished job 0.
33 of 33 steps (100%) done
Complete log: /home/alper/tmp/snakemake-dna-seq-gatk-variant-calling-1.0.1/.snakemake/log/2019-04-29T103714.426290.snakemake.log</p></div>
</body>
</html>